// @peerline/hogak-player/dist/index-Dq-62_9z.js
import { jsx as me, jsxs as eo } from "react/jsx-runtime";
import ye, { useRef as Wr, useDebugValue as cr, createElement as to, useContext as ro, useState as Oe } from "react";
import { PlayTime as lr } from "./components/PlayTime/index.js";
var Pt = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function no(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
function oo(e) {
  if (e.__esModule)
    return e;
  var t = e.default;
  if (typeof t == "function") {
    var r = function n() {
      return this instanceof n ? Reflect.construct(t, arguments, this.constructor) : t.apply(this, arguments);
    };
    r.prototype = t.prototype;
  } else
    r = {};
  return Object.defineProperty(r, "__esModule", { value: !0 }), Object.keys(e).forEach(function(n) {
    var o = Object.getOwnPropertyDescriptor(e, n);
    Object.defineProperty(r, n, o.get ? o : {
      enumerable: !0,
      get: function() {
        return e[n];
      }
    });
  }), r;
}
var ao = function(t, r, n) {
  var o = document.head || document.getElementsByTagName("head")[0], a = document.createElement("script");
  typeof r == "function" && (n = r, r = {}), r = r || {}, n = n || function() {
  }, a.type = r.type || "text/javascript", a.charset = r.charset || "utf8", a.async = "async" in r ? !!r.async : !0, a.src = t, r.attrs && io(a, r.attrs), r.text && (a.text = "" + r.text);
  var i = "onload" in a ? ur : so;
  i(a, n), a.onload || ur(a, n), o.appendChild(a);
};
function io(e, t) {
  for (var r in t)
    e.setAttribute(r, t[r]);
}
function ur(e, t) {
  e.onload = function() {
    this.onerror = this.onload = null, t(null, e);
  }, e.onerror = function() {
    this.onerror = this.onload = null, t(new Error("Failed to load " + this.src), e);
  };
}
function so(e, t) {
  e.onreadystatechange = function() {
    this.readyState != "complete" && this.readyState != "loaded" || (this.onreadystatechange = null, t(null, e));
  };
}
var co = function(t) {
  return lo(t) && !uo(t);
};
function lo(e) {
  return !!e && typeof e == "object";
}
function uo(e) {
  var t = Object.prototype.toString.call(e);
  return t === "[object RegExp]" || t === "[object Date]" || yo(e);
}
var fo = typeof Symbol == "function" && Symbol.for, po = fo ? Symbol.for("react.element") : 60103;
function yo(e) {
  return e.$$typeof === po;
}
function ho(e) {
  return Array.isArray(e) ? [] : {};
}
function We(e, t) {
  return t.clone !== !1 && t.isMergeableObject(e) ? Re(ho(e), e, t) : e;
}
function mo(e, t, r) {
  return e.concat(t).map(function(n) {
    return We(n, r);
  });
}
function go(e, t) {
  if (!t.customMerge)
    return Re;
  var r = t.customMerge(e);
  return typeof r == "function" ? r : Re;
}
function vo(e) {
  return Object.getOwnPropertySymbols ? Object.getOwnPropertySymbols(e).filter(function(t) {
    return Object.propertyIsEnumerable.call(e, t);
  }) : [];
}
function fr(e) {
  return Object.keys(e).concat(vo(e));
}
function Br(e, t) {
  try {
    return t in e;
  } catch {
    return !1;
  }
}
function bo(e, t) {
  return Br(e, t) && !(Object.hasOwnProperty.call(e, t) && Object.propertyIsEnumerable.call(e, t));
}
function _o(e, t, r) {
  var n = {};
  return r.isMergeableObject(e) && fr(e).forEach(function(o) {
    n[o] = We(e[o], r);
  }), fr(t).forEach(function(o) {
    bo(e, o) || (Br(e, o) && r.isMergeableObject(t[o]) ? n[o] = go(o, r)(e[o], t[o], r) : n[o] = We(t[o], r));
  }), n;
}
function Re(e, t, r) {
  r = r || {}, r.arrayMerge = r.arrayMerge || mo, r.isMergeableObject = r.isMergeableObject || co, r.cloneUnlessOtherwiseSpecified = We;
  var n = Array.isArray(t), o = Array.isArray(e), a = n === o;
  return a ? n ? r.arrayMerge(e, t, r) : _o(e, t, r) : We(t, r);
}
Re.all = function(t, r) {
  if (!Array.isArray(t))
    throw new Error("first argument should be an array");
  return t.reduce(function(n, o) {
    return Re(n, o, r);
  }, {});
};
var Po = Re, qr = Po, wo = Object.create, ut = Object.defineProperty, So = Object.getOwnPropertyDescriptor, Oo = Object.getOwnPropertyNames, Eo = Object.getPrototypeOf, To = Object.prototype.hasOwnProperty, Co = (e, t) => {
  for (var r in t)
    ut(e, r, { get: t[r], enumerable: !0 });
}, Xr = (e, t, r, n) => {
  if (t && typeof t == "object" || typeof t == "function")
    for (let o of Oo(t))
      !To.call(e, o) && o !== r && ut(e, o, { get: () => t[o], enumerable: !(n = So(t, o)) || n.enumerable });
  return e;
}, Xt = (e, t, r) => (r = e != null ? wo(Eo(e)) : {}, Xr(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  t || !e || !e.__esModule ? ut(r, "default", { value: e, enumerable: !0 }) : r,
  e
)), Ao = (e) => Xr(ut({}, "__esModule", { value: !0 }), e), Gr = {};
Co(Gr, {
  callPlayer: () => Yo,
  getConfig: () => Fo,
  getSDK: () => Ho,
  isBlobUrl: () => Bo,
  isMediaStream: () => Wo,
  lazy: () => Io,
  omit: () => Vo,
  parseEndTime: () => Lo,
  parseStartTime: () => No,
  queryString: () => zo,
  randomString: () => Uo,
  supportsWebKitPresentationMode: () => qo
});
var ft = Ao(Gr), Ro = Xt(ye), xo = Xt(ao), ko = Xt(qr);
const Io = (e) => Ro.default.lazy(async () => {
  const t = await e();
  return typeof t.default == "function" ? t : t.default;
}), Mo = /[?&#](?:start|t)=([0-9hms]+)/, $o = /[?&#]end=([0-9hms]+)/, jt = /(\d+)(h|m|s)/g, jo = /^\d+$/;
function Kr(e, t) {
  if (e instanceof Array)
    return;
  const r = e.match(t);
  if (r) {
    const n = r[1];
    if (n.match(jt))
      return Do(n);
    if (jo.test(n))
      return parseInt(n);
  }
}
function Do(e) {
  let t = 0, r = jt.exec(e);
  for (; r !== null; ) {
    const [, n, o] = r;
    o === "h" && (t += parseInt(n, 10) * 60 * 60), o === "m" && (t += parseInt(n, 10) * 60), o === "s" && (t += parseInt(n, 10)), r = jt.exec(e);
  }
  return t;
}
function No(e) {
  return Kr(e, Mo);
}
function Lo(e) {
  return Kr(e, $o);
}
function Uo() {
  return Math.random().toString(36).substr(2, 5);
}
function zo(e) {
  return Object.keys(e).map((t) => `${t}=${e[t]}`).join("&");
}
function wt(e) {
  return window[e] ? window[e] : window.exports && window.exports[e] ? window.exports[e] : window.module && window.module.exports && window.module.exports[e] ? window.module.exports[e] : null;
}
const Ee = {}, Ho = function(t, r, n = null, o = () => !0, a = xo.default) {
  const i = wt(r);
  return i && o(i) ? Promise.resolve(i) : new Promise((s, l) => {
    if (Ee[t]) {
      Ee[t].push({ resolve: s, reject: l });
      return;
    }
    Ee[t] = [{ resolve: s, reject: l }];
    const h = (p) => {
      Ee[t].forEach((d) => d.resolve(p));
    };
    if (n) {
      const p = window[n];
      window[n] = function() {
        p && p(), h(wt(r));
      };
    }
    a(t, (p) => {
      p ? (Ee[t].forEach((d) => d.reject(p)), Ee[t] = null) : n || h(wt(r));
    });
  });
};
function Fo(e, t) {
  return (0, ko.default)(t.config, e.config);
}
function Vo(e, ...t) {
  const r = [].concat(...t), n = {}, o = Object.keys(e);
  for (const a of o)
    r.indexOf(a) === -1 && (n[a] = e[a]);
  return n;
}
function Yo(e, ...t) {
  if (!this.player || !this.player[e]) {
    let r = `ReactPlayer: ${this.constructor.displayName} player could not call %c${e}%c – `;
    return this.player ? this.player[e] || (r += "The method was not available") : r += "The player was not available", console.warn(r, "font-weight: bold", ""), null;
  }
  return this.player[e](...t);
}
function Wo(e) {
  return typeof window < "u" && typeof window.MediaStream < "u" && e instanceof window.MediaStream;
}
function Bo(e) {
  return /^blob:/.test(e);
}
function qo(e = document.createElement("video")) {
  const t = /iPhone|iPod/.test(navigator.userAgent) === !1;
  return e.webkitSupportsPresentationMode && typeof e.webkitSetPresentationMode == "function" && t;
}
var Gt = Object.defineProperty, Xo = Object.getOwnPropertyDescriptor, Go = Object.getOwnPropertyNames, Ko = Object.prototype.hasOwnProperty, Jo = (e, t) => {
  for (var r in t)
    Gt(e, r, { get: t[r], enumerable: !0 });
}, Zo = (e, t, r, n) => {
  if (t && typeof t == "object" || typeof t == "function")
    for (let o of Go(t))
      !Ko.call(e, o) && o !== r && Gt(e, o, { get: () => t[o], enumerable: !(n = Xo(t, o)) || n.enumerable });
  return e;
}, Qo = (e) => Zo(Gt({}, "__esModule", { value: !0 }), e), Jr = {};
Jo(Jr, {
  AUDIO_EXTENSIONS: () => Kt,
  DASH_EXTENSIONS: () => pn,
  FLV_EXTENSIONS: () => dn,
  HLS_EXTENSIONS: () => Zt,
  MATCH_URL_DAILYMOTION: () => cn,
  MATCH_URL_FACEBOOK: () => tn,
  MATCH_URL_FACEBOOK_WATCH: () => rn,
  MATCH_URL_KALTURA: () => fn,
  MATCH_URL_MIXCLOUD: () => ln,
  MATCH_URL_MUX: () => en,
  MATCH_URL_SOUNDCLOUD: () => Zr,
  MATCH_URL_STREAMABLE: () => nn,
  MATCH_URL_TWITCH_CHANNEL: () => sn,
  MATCH_URL_TWITCH_VIDEO: () => an,
  MATCH_URL_VIDYARD: () => un,
  MATCH_URL_VIMEO: () => Qr,
  MATCH_URL_WISTIA: () => on,
  MATCH_URL_YOUTUBE: () => Dt,
  VIDEO_EXTENSIONS: () => Jt,
  canPlay: () => ta
});
var ea = Qo(Jr), pr = ft;
const Dt = /(?:youtu\.be\/|youtube(?:-nocookie|education)?\.com\/(?:embed\/|v\/|watch\/|watch\?v=|watch\?.+&v=|shorts\/|live\/))((\w|-){11})|youtube\.com\/playlist\?list=|youtube\.com\/user\//, Zr = /(?:soundcloud\.com|snd\.sc)\/[^.]+$/, Qr = /vimeo\.com\/(?!progressive_redirect).+/, en = /stream\.mux\.com\/(?!\w+\.m3u8)(\w+)/, tn = /^https?:\/\/(www\.)?facebook\.com.*\/(video(s)?|watch|story)(\.php?|\/).+$/, rn = /^https?:\/\/fb\.watch\/.+$/, nn = /streamable\.com\/([a-z0-9]+)$/, on = /(?:wistia\.(?:com|net)|wi\.st)\/(?:medias|embed)\/(?:iframe\/)?([^?]+)/, an = /(?:www\.|go\.)?twitch\.tv\/videos\/(\d+)($|\?)/, sn = /(?:www\.|go\.)?twitch\.tv\/([a-zA-Z0-9_]+)($|\?)/, cn = /^(?:(?:https?):)?(?:\/\/)?(?:www\.)?(?:(?:dailymotion\.com(?:\/embed)?\/video)|dai\.ly)\/([a-zA-Z0-9]+)(?:_[\w_-]+)?(?:[\w.#_-]+)?/, ln = /mixcloud\.com\/([^/]+\/[^/]+)/, un = /vidyard.com\/(?:watch\/)?([a-zA-Z0-9-_]+)/, fn = /^https?:\/\/[a-zA-Z]+\.kaltura.(com|org)\/p\/([0-9]+)\/sp\/([0-9]+)00\/embedIframeJs\/uiconf_id\/([0-9]+)\/partner_id\/([0-9]+)(.*)entry_id.([a-zA-Z0-9-_].*)$/, Kt = /\.(m4a|m4b|mp4a|mpga|mp2|mp2a|mp3|m2a|m3a|wav|weba|aac|oga|spx)($|\?)/i, Jt = /\.(mp4|og[gv]|webm|mov|m4v)(#t=[,\d+]+)?($|\?)/i, Zt = /\.(m3u8)($|\?)/i, pn = /\.(mpd)($|\?)/i, dn = /\.(flv)($|\?)/i, Nt = (e) => {
  if (e instanceof Array) {
    for (const t of e)
      if (typeof t == "string" && Nt(t) || Nt(t.src))
        return !0;
    return !1;
  }
  return (0, pr.isMediaStream)(e) || (0, pr.isBlobUrl)(e) ? !0 : Kt.test(e) || Jt.test(e) || Zt.test(e) || pn.test(e) || dn.test(e);
}, ta = {
  youtube: (e) => e instanceof Array ? e.every((t) => Dt.test(t)) : Dt.test(e),
  soundcloud: (e) => Zr.test(e) && !Kt.test(e),
  vimeo: (e) => Qr.test(e) && !Jt.test(e) && !Zt.test(e),
  mux: (e) => en.test(e),
  facebook: (e) => tn.test(e) || rn.test(e),
  streamable: (e) => nn.test(e),
  wistia: (e) => on.test(e),
  twitch: (e) => an.test(e) || sn.test(e),
  dailymotion: (e) => cn.test(e),
  mixcloud: (e) => ln.test(e),
  vidyard: (e) => un.test(e),
  kaltura: (e) => fn.test(e),
  file: Nt
};
var Qt = Object.defineProperty, ra = Object.getOwnPropertyDescriptor, na = Object.getOwnPropertyNames, oa = Object.prototype.hasOwnProperty, aa = (e, t) => {
  for (var r in t)
    Qt(e, r, { get: t[r], enumerable: !0 });
}, ia = (e, t, r, n) => {
  if (t && typeof t == "object" || typeof t == "function")
    for (let o of na(t))
      !oa.call(e, o) && o !== r && Qt(e, o, { get: () => t[o], enumerable: !(n = ra(t, o)) || n.enumerable });
  return e;
}, sa = (e) => ia(Qt({}, "__esModule", { value: !0 }), e), yn = {};
aa(yn, {
  default: () => la
});
var ca = sa(yn), Q = ft, J = ea, la = [
  {
    key: "youtube",
    name: "YouTube",
    canPlay: J.canPlay.youtube,
    lazyPlayer: (0, Q.lazy)(() => import(
      /* webpackChunkName: 'reactPlayerYouTube' */
      "./YouTube-CdPxmy_i.js"
    ).then((e) => e.Y))
  },
  {
    key: "soundcloud",
    name: "SoundCloud",
    canPlay: J.canPlay.soundcloud,
    lazyPlayer: (0, Q.lazy)(() => import(
      /* webpackChunkName: 'reactPlayerSoundCloud' */
      "./SoundCloud-btWnFAz9.js"
    ).then((e) => e.S))
  },
  {
    key: "vimeo",
    name: "Vimeo",
    canPlay: J.canPlay.vimeo,
    lazyPlayer: (0, Q.lazy)(() => import(
      /* webpackChunkName: 'reactPlayerVimeo' */
      "./Vimeo-DlI7Symo.js"
    ).then((e) => e.V))
  },
  {
    key: "mux",
    name: "Mux",
    canPlay: J.canPlay.mux,
    lazyPlayer: (0, Q.lazy)(() => import(
      /* webpackChunkName: 'reactPlayerMux' */
      "./Mux-C4-RRZ1k.js"
    ).then((e) => e.M))
  },
  {
    key: "facebook",
    name: "Facebook",
    canPlay: J.canPlay.facebook,
    lazyPlayer: (0, Q.lazy)(() => import(
      /* webpackChunkName: 'reactPlayerFacebook' */
      "./Facebook-CLdmwvgF.js"
    ).then((e) => e.F))
  },
  {
    key: "streamable",
    name: "Streamable",
    canPlay: J.canPlay.streamable,
    lazyPlayer: (0, Q.lazy)(() => import(
      /* webpackChunkName: 'reactPlayerStreamable' */
      "./Streamable-CHdo90Rc.js"
    ).then((e) => e.S))
  },
  {
    key: "wistia",
    name: "Wistia",
    canPlay: J.canPlay.wistia,
    lazyPlayer: (0, Q.lazy)(() => import(
      /* webpackChunkName: 'reactPlayerWistia' */
      "./Wistia-DxlJQBDZ.js"
    ).then((e) => e.W))
  },
  {
    key: "twitch",
    name: "Twitch",
    canPlay: J.canPlay.twitch,
    lazyPlayer: (0, Q.lazy)(() => import(
      /* webpackChunkName: 'reactPlayerTwitch' */
      "./Twitch-CsG73pAp.js"
    ).then((e) => e.T))
  },
  {
    key: "dailymotion",
    name: "DailyMotion",
    canPlay: J.canPlay.dailymotion,
    lazyPlayer: (0, Q.lazy)(() => import(
      /* webpackChunkName: 'reactPlayerDailyMotion' */
      "./DailyMotion-C63ROQ9o.js"
    ).then((e) => e.D))
  },
  {
    key: "mixcloud",
    name: "Mixcloud",
    canPlay: J.canPlay.mixcloud,
    lazyPlayer: (0, Q.lazy)(() => import(
      /* webpackChunkName: 'reactPlayerMixcloud' */
      "./Mixcloud-CgDGx3ed.js"
    ).then((e) => e.M))
  },
  {
    key: "vidyard",
    name: "Vidyard",
    canPlay: J.canPlay.vidyard,
    lazyPlayer: (0, Q.lazy)(() => import(
      /* webpackChunkName: 'reactPlayerVidyard' */
      "./Vidyard-ea0xkaa8.js"
    ).then((e) => e.V))
  },
  {
    key: "kaltura",
    name: "Kaltura",
    canPlay: J.canPlay.kaltura,
    lazyPlayer: (0, Q.lazy)(() => import(
      /* webpackChunkName: 'reactPlayerKaltura' */
      "./Kaltura-dGyHYaBN.js"
    ).then((e) => e.K))
  },
  {
    key: "file",
    name: "FilePlayer",
    canPlay: J.canPlay.file,
    canEnablePIP: (e) => J.canPlay.file(e) && (document.pictureInPictureEnabled || (0, Q.supportsWebKitPresentationMode)()) && !J.AUDIO_EXTENSIONS.test(e),
    lazyPlayer: (0, Q.lazy)(() => import(
      /* webpackChunkName: 'reactPlayerFilePlayer' */
      "./FilePlayer-lHWQc_ex.js"
    ).then((e) => e.F))
  }
], dr = Number.isNaN || function(t) {
  return typeof t == "number" && t !== t;
};
function ua(e, t) {
  return !!(e === t || dr(e) && dr(t));
}
function fa(e, t) {
  if (e.length !== t.length)
    return !1;
  for (var r = 0; r < e.length; r++)
    if (!ua(e[r], t[r]))
      return !1;
  return !0;
}
function pa(e, t) {
  t === void 0 && (t = fa);
  var r, n = [], o, a = !1;
  function i() {
    for (var s = [], l = 0; l < arguments.length; l++)
      s[l] = arguments[l];
    return a && r === this && t(s, n) || (o = e.apply(this, s), a = !0, r = this, n = s), o;
  }
  return i;
}
const da = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: pa
}, Symbol.toStringTag, { value: "Module" })), ya = /* @__PURE__ */ oo(da);
var ha = typeof Element < "u", ma = typeof Map == "function", ga = typeof Set == "function", va = typeof ArrayBuffer == "function" && !!ArrayBuffer.isView;
function tt(e, t) {
  if (e === t)
    return !0;
  if (e && t && typeof e == "object" && typeof t == "object") {
    if (e.constructor !== t.constructor)
      return !1;
    var r, n, o;
    if (Array.isArray(e)) {
      if (r = e.length, r != t.length)
        return !1;
      for (n = r; n-- !== 0; )
        if (!tt(e[n], t[n]))
          return !1;
      return !0;
    }
    var a;
    if (ma && e instanceof Map && t instanceof Map) {
      if (e.size !== t.size)
        return !1;
      for (a = e.entries(); !(n = a.next()).done; )
        if (!t.has(n.value[0]))
          return !1;
      for (a = e.entries(); !(n = a.next()).done; )
        if (!tt(n.value[1], t.get(n.value[0])))
          return !1;
      return !0;
    }
    if (ga && e instanceof Set && t instanceof Set) {
      if (e.size !== t.size)
        return !1;
      for (a = e.entries(); !(n = a.next()).done; )
        if (!t.has(n.value[0]))
          return !1;
      return !0;
    }
    if (va && ArrayBuffer.isView(e) && ArrayBuffer.isView(t)) {
      if (r = e.length, r != t.length)
        return !1;
      for (n = r; n-- !== 0; )
        if (e[n] !== t[n])
          return !1;
      return !0;
    }
    if (e.constructor === RegExp)
      return e.source === t.source && e.flags === t.flags;
    if (e.valueOf !== Object.prototype.valueOf && typeof e.valueOf == "function" && typeof t.valueOf == "function")
      return e.valueOf() === t.valueOf();
    if (e.toString !== Object.prototype.toString && typeof e.toString == "function" && typeof t.toString == "function")
      return e.toString() === t.toString();
    if (o = Object.keys(e), r = o.length, r !== Object.keys(t).length)
      return !1;
    for (n = r; n-- !== 0; )
      if (!Object.prototype.hasOwnProperty.call(t, o[n]))
        return !1;
    if (ha && e instanceof Element)
      return !1;
    for (n = r; n-- !== 0; )
      if (!((o[n] === "_owner" || o[n] === "__v" || o[n] === "__o") && e.$$typeof) && !tt(e[o[n]], t[o[n]]))
        return !1;
    return !0;
  }
  return e !== e && t !== t;
}
var hn = function(t, r) {
  try {
    return tt(t, r);
  } catch (n) {
    if ((n.message || "").match(/stack|recursion/i))
      return console.warn("react-fast-compare cannot handle circular refs"), !1;
    throw n;
  }
}, Lt = { exports: {} }, Ke = { exports: {} }, N = {};
/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var yr;
function ba() {
  if (yr)
    return N;
  yr = 1;
  var e = typeof Symbol == "function" && Symbol.for, t = e ? Symbol.for("react.element") : 60103, r = e ? Symbol.for("react.portal") : 60106, n = e ? Symbol.for("react.fragment") : 60107, o = e ? Symbol.for("react.strict_mode") : 60108, a = e ? Symbol.for("react.profiler") : 60114, i = e ? Symbol.for("react.provider") : 60109, s = e ? Symbol.for("react.context") : 60110, l = e ? Symbol.for("react.async_mode") : 60111, h = e ? Symbol.for("react.concurrent_mode") : 60111, p = e ? Symbol.for("react.forward_ref") : 60112, d = e ? Symbol.for("react.suspense") : 60113, m = e ? Symbol.for("react.suspense_list") : 60120, g = e ? Symbol.for("react.memo") : 60115, R = e ? Symbol.for("react.lazy") : 60116, b = e ? Symbol.for("react.block") : 60121, j = e ? Symbol.for("react.fundamental") : 60117, $ = e ? Symbol.for("react.responder") : 60118, D = e ? Symbol.for("react.scope") : 60119;
  function _(u) {
    if (typeof u == "object" && u !== null) {
      var O = u.$$typeof;
      switch (O) {
        case t:
          switch (u = u.type, u) {
            case l:
            case h:
            case n:
            case a:
            case o:
            case d:
              return u;
            default:
              switch (u = u && u.$$typeof, u) {
                case s:
                case p:
                case R:
                case g:
                case i:
                  return u;
                default:
                  return O;
              }
          }
        case r:
          return O;
      }
    }
  }
  function P(u) {
    return _(u) === h;
  }
  return N.AsyncMode = l, N.ConcurrentMode = h, N.ContextConsumer = s, N.ContextProvider = i, N.Element = t, N.ForwardRef = p, N.Fragment = n, N.Lazy = R, N.Memo = g, N.Portal = r, N.Profiler = a, N.StrictMode = o, N.Suspense = d, N.isAsyncMode = function(u) {
    return P(u) || _(u) === l;
  }, N.isConcurrentMode = P, N.isContextConsumer = function(u) {
    return _(u) === s;
  }, N.isContextProvider = function(u) {
    return _(u) === i;
  }, N.isElement = function(u) {
    return typeof u == "object" && u !== null && u.$$typeof === t;
  }, N.isForwardRef = function(u) {
    return _(u) === p;
  }, N.isFragment = function(u) {
    return _(u) === n;
  }, N.isLazy = function(u) {
    return _(u) === R;
  }, N.isMemo = function(u) {
    return _(u) === g;
  }, N.isPortal = function(u) {
    return _(u) === r;
  }, N.isProfiler = function(u) {
    return _(u) === a;
  }, N.isStrictMode = function(u) {
    return _(u) === o;
  }, N.isSuspense = function(u) {
    return _(u) === d;
  }, N.isValidElementType = function(u) {
    return typeof u == "string" || typeof u == "function" || u === n || u === h || u === a || u === o || u === d || u === m || typeof u == "object" && u !== null && (u.$$typeof === R || u.$$typeof === g || u.$$typeof === i || u.$$typeof === s || u.$$typeof === p || u.$$typeof === j || u.$$typeof === $ || u.$$typeof === D || u.$$typeof === b);
  }, N.typeOf = _, N;
}
var L = {};
/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var hr;
function _a() {
  return hr || (hr = 1, process.env.NODE_ENV !== "production" && function() {
    var e = typeof Symbol == "function" && Symbol.for, t = e ? Symbol.for("react.element") : 60103, r = e ? Symbol.for("react.portal") : 60106, n = e ? Symbol.for("react.fragment") : 60107, o = e ? Symbol.for("react.strict_mode") : 60108, a = e ? Symbol.for("react.profiler") : 60114, i = e ? Symbol.for("react.provider") : 60109, s = e ? Symbol.for("react.context") : 60110, l = e ? Symbol.for("react.async_mode") : 60111, h = e ? Symbol.for("react.concurrent_mode") : 60111, p = e ? Symbol.for("react.forward_ref") : 60112, d = e ? Symbol.for("react.suspense") : 60113, m = e ? Symbol.for("react.suspense_list") : 60120, g = e ? Symbol.for("react.memo") : 60115, R = e ? Symbol.for("react.lazy") : 60116, b = e ? Symbol.for("react.block") : 60121, j = e ? Symbol.for("react.fundamental") : 60117, $ = e ? Symbol.for("react.responder") : 60118, D = e ? Symbol.for("react.scope") : 60119;
    function _(f) {
      return typeof f == "string" || typeof f == "function" || // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
      f === n || f === h || f === a || f === o || f === d || f === m || typeof f == "object" && f !== null && (f.$$typeof === R || f.$$typeof === g || f.$$typeof === i || f.$$typeof === s || f.$$typeof === p || f.$$typeof === j || f.$$typeof === $ || f.$$typeof === D || f.$$typeof === b);
    }
    function P(f) {
      if (typeof f == "object" && f !== null) {
        var le = f.$$typeof;
        switch (le) {
          case t:
            var Ge = f.type;
            switch (Ge) {
              case l:
              case h:
              case n:
              case a:
              case o:
              case d:
                return Ge;
              default:
                var sr = Ge && Ge.$$typeof;
                switch (sr) {
                  case s:
                  case p:
                  case R:
                  case g:
                  case i:
                    return sr;
                  default:
                    return le;
                }
            }
          case r:
            return le;
        }
      }
    }
    var u = l, O = h, w = s, ne = i, ie = t, q = p, we = n, Le = R, Se = g, pe = r, Ue = a, oe = o, se = d, he = !1;
    function te(f) {
      return he || (he = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.")), c(f) || P(f) === l;
    }
    function c(f) {
      return P(f) === h;
    }
    function y(f) {
      return P(f) === s;
    }
    function v(f) {
      return P(f) === i;
    }
    function C(f) {
      return typeof f == "object" && f !== null && f.$$typeof === t;
    }
    function S(f) {
      return P(f) === p;
    }
    function k(f) {
      return P(f) === n;
    }
    function E(f) {
      return P(f) === R;
    }
    function T(f) {
      return P(f) === g;
    }
    function x(f) {
      return P(f) === r;
    }
    function I(f) {
      return P(f) === a;
    }
    function M(f) {
      return P(f) === o;
    }
    function Y(f) {
      return P(f) === d;
    }
    L.AsyncMode = u, L.ConcurrentMode = O, L.ContextConsumer = w, L.ContextProvider = ne, L.Element = ie, L.ForwardRef = q, L.Fragment = we, L.Lazy = Le, L.Memo = Se, L.Portal = pe, L.Profiler = Ue, L.StrictMode = oe, L.Suspense = se, L.isAsyncMode = te, L.isConcurrentMode = c, L.isContextConsumer = y, L.isContextProvider = v, L.isElement = C, L.isForwardRef = S, L.isFragment = k, L.isLazy = E, L.isMemo = T, L.isPortal = x, L.isProfiler = I, L.isStrictMode = M, L.isSuspense = Y, L.isValidElementType = _, L.typeOf = P;
  }()), L;
}
var mr;
function mn() {
  return mr || (mr = 1, process.env.NODE_ENV === "production" ? Ke.exports = ba() : Ke.exports = _a()), Ke.exports;
}
/*
object-assign
(c) Sindre Sorhus
@license MIT
*/
var St, gr;
function Pa() {
  if (gr)
    return St;
  gr = 1;
  var e = Object.getOwnPropertySymbols, t = Object.prototype.hasOwnProperty, r = Object.prototype.propertyIsEnumerable;
  function n(a) {
    if (a == null)
      throw new TypeError("Object.assign cannot be called with null or undefined");
    return Object(a);
  }
  function o() {
    try {
      if (!Object.assign)
        return !1;
      var a = new String("abc");
      if (a[5] = "de", Object.getOwnPropertyNames(a)[0] === "5")
        return !1;
      for (var i = {}, s = 0; s < 10; s++)
        i["_" + String.fromCharCode(s)] = s;
      var l = Object.getOwnPropertyNames(i).map(function(p) {
        return i[p];
      });
      if (l.join("") !== "0123456789")
        return !1;
      var h = {};
      return "abcdefghijklmnopqrst".split("").forEach(function(p) {
        h[p] = p;
      }), Object.keys(Object.assign({}, h)).join("") === "abcdefghijklmnopqrst";
    } catch {
      return !1;
    }
  }
  return St = o() ? Object.assign : function(a, i) {
    for (var s, l = n(a), h, p = 1; p < arguments.length; p++) {
      s = Object(arguments[p]);
      for (var d in s)
        t.call(s, d) && (l[d] = s[d]);
      if (e) {
        h = e(s);
        for (var m = 0; m < h.length; m++)
          r.call(s, h[m]) && (l[h[m]] = s[h[m]]);
      }
    }
    return l;
  }, St;
}
var Ot, vr;
function er() {
  if (vr)
    return Ot;
  vr = 1;
  var e = "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED";
  return Ot = e, Ot;
}
var Et, br;
function gn() {
  return br || (br = 1, Et = Function.call.bind(Object.prototype.hasOwnProperty)), Et;
}
var Tt, _r;
function wa() {
  if (_r)
    return Tt;
  _r = 1;
  var e = function() {
  };
  if (process.env.NODE_ENV !== "production") {
    var t = er(), r = {}, n = gn();
    e = function(a) {
      var i = "Warning: " + a;
      typeof console < "u" && console.error(i);
      try {
        throw new Error(i);
      } catch {
      }
    };
  }
  function o(a, i, s, l, h) {
    if (process.env.NODE_ENV !== "production") {
      for (var p in a)
        if (n(a, p)) {
          var d;
          try {
            if (typeof a[p] != "function") {
              var m = Error(
                (l || "React class") + ": " + s + " type `" + p + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof a[p] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`."
              );
              throw m.name = "Invariant Violation", m;
            }
            d = a[p](i, p, l, s, null, t);
          } catch (R) {
            d = R;
          }
          if (d && !(d instanceof Error) && e(
            (l || "React class") + ": type specification of " + s + " `" + p + "` is invalid; the type checker function must return `null` or an `Error` but returned a " + typeof d + ". You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument)."
          ), d instanceof Error && !(d.message in r)) {
            r[d.message] = !0;
            var g = h ? h() : "";
            e(
              "Failed " + s + " type: " + d.message + (g ?? "")
            );
          }
        }
    }
  }
  return o.resetWarningCache = function() {
    process.env.NODE_ENV !== "production" && (r = {});
  }, Tt = o, Tt;
}
var Ct, Pr;
function Sa() {
  if (Pr)
    return Ct;
  Pr = 1;
  var e = mn(), t = Pa(), r = er(), n = gn(), o = wa(), a = function() {
  };
  process.env.NODE_ENV !== "production" && (a = function(s) {
    var l = "Warning: " + s;
    typeof console < "u" && console.error(l);
    try {
      throw new Error(l);
    } catch {
    }
  });
  function i() {
    return null;
  }
  return Ct = function(s, l) {
    var h = typeof Symbol == "function" && Symbol.iterator, p = "@@iterator";
    function d(c) {
      var y = c && (h && c[h] || c[p]);
      if (typeof y == "function")
        return y;
    }
    var m = "<<anonymous>>", g = {
      array: $("array"),
      bigint: $("bigint"),
      bool: $("boolean"),
      func: $("function"),
      number: $("number"),
      object: $("object"),
      string: $("string"),
      symbol: $("symbol"),
      any: D(),
      arrayOf: _,
      element: P(),
      elementType: u(),
      instanceOf: O,
      node: q(),
      objectOf: ne,
      oneOf: w,
      oneOfType: ie,
      shape: Le,
      exact: Se
    };
    function R(c, y) {
      return c === y ? c !== 0 || 1 / c === 1 / y : c !== c && y !== y;
    }
    function b(c, y) {
      this.message = c, this.data = y && typeof y == "object" ? y : {}, this.stack = "";
    }
    b.prototype = Error.prototype;
    function j(c) {
      if (process.env.NODE_ENV !== "production")
        var y = {}, v = 0;
      function C(k, E, T, x, I, M, Y) {
        if (x = x || m, M = M || T, Y !== r) {
          if (l) {
            var f = new Error(
              "Calling PropTypes validators directly is not supported by the `prop-types` package. Use `PropTypes.checkPropTypes()` to call them. Read more at http://fb.me/use-check-prop-types"
            );
            throw f.name = "Invariant Violation", f;
          } else if (process.env.NODE_ENV !== "production" && typeof console < "u") {
            var le = x + ":" + T;
            !y[le] && // Avoid spamming the console because they are often not actionable except for lib authors
            v < 3 && (a(
              "You are manually calling a React.PropTypes validation function for the `" + M + "` prop on `" + x + "`. This is deprecated and will throw in the standalone `prop-types` package. You may be seeing this warning due to a third-party PropTypes library. See https://fb.me/react-warning-dont-call-proptypes for details."
            ), y[le] = !0, v++);
          }
        }
        return E[T] == null ? k ? E[T] === null ? new b("The " + I + " `" + M + "` is marked as required " + ("in `" + x + "`, but its value is `null`.")) : new b("The " + I + " `" + M + "` is marked as required in " + ("`" + x + "`, but its value is `undefined`.")) : null : c(E, T, x, I, M);
      }
      var S = C.bind(null, !1);
      return S.isRequired = C.bind(null, !0), S;
    }
    function $(c) {
      function y(v, C, S, k, E, T) {
        var x = v[C], I = oe(x);
        if (I !== c) {
          var M = se(x);
          return new b(
            "Invalid " + k + " `" + E + "` of type " + ("`" + M + "` supplied to `" + S + "`, expected ") + ("`" + c + "`."),
            { expectedType: c }
          );
        }
        return null;
      }
      return j(y);
    }
    function D() {
      return j(i);
    }
    function _(c) {
      function y(v, C, S, k, E) {
        if (typeof c != "function")
          return new b("Property `" + E + "` of component `" + S + "` has invalid PropType notation inside arrayOf.");
        var T = v[C];
        if (!Array.isArray(T)) {
          var x = oe(T);
          return new b("Invalid " + k + " `" + E + "` of type " + ("`" + x + "` supplied to `" + S + "`, expected an array."));
        }
        for (var I = 0; I < T.length; I++) {
          var M = c(T, I, S, k, E + "[" + I + "]", r);
          if (M instanceof Error)
            return M;
        }
        return null;
      }
      return j(y);
    }
    function P() {
      function c(y, v, C, S, k) {
        var E = y[v];
        if (!s(E)) {
          var T = oe(E);
          return new b("Invalid " + S + " `" + k + "` of type " + ("`" + T + "` supplied to `" + C + "`, expected a single ReactElement."));
        }
        return null;
      }
      return j(c);
    }
    function u() {
      function c(y, v, C, S, k) {
        var E = y[v];
        if (!e.isValidElementType(E)) {
          var T = oe(E);
          return new b("Invalid " + S + " `" + k + "` of type " + ("`" + T + "` supplied to `" + C + "`, expected a single ReactElement type."));
        }
        return null;
      }
      return j(c);
    }
    function O(c) {
      function y(v, C, S, k, E) {
        if (!(v[C] instanceof c)) {
          var T = c.name || m, x = te(v[C]);
          return new b("Invalid " + k + " `" + E + "` of type " + ("`" + x + "` supplied to `" + S + "`, expected ") + ("instance of `" + T + "`."));
        }
        return null;
      }
      return j(y);
    }
    function w(c) {
      if (!Array.isArray(c))
        return process.env.NODE_ENV !== "production" && (arguments.length > 1 ? a(
          "Invalid arguments supplied to oneOf, expected an array, got " + arguments.length + " arguments. A common mistake is to write oneOf(x, y, z) instead of oneOf([x, y, z])."
        ) : a("Invalid argument supplied to oneOf, expected an array.")), i;
      function y(v, C, S, k, E) {
        for (var T = v[C], x = 0; x < c.length; x++)
          if (R(T, c[x]))
            return null;
        var I = JSON.stringify(c, function(Y, f) {
          var le = se(f);
          return le === "symbol" ? String(f) : f;
        });
        return new b("Invalid " + k + " `" + E + "` of value `" + String(T) + "` " + ("supplied to `" + S + "`, expected one of " + I + "."));
      }
      return j(y);
    }
    function ne(c) {
      function y(v, C, S, k, E) {
        if (typeof c != "function")
          return new b("Property `" + E + "` of component `" + S + "` has invalid PropType notation inside objectOf.");
        var T = v[C], x = oe(T);
        if (x !== "object")
          return new b("Invalid " + k + " `" + E + "` of type " + ("`" + x + "` supplied to `" + S + "`, expected an object."));
        for (var I in T)
          if (n(T, I)) {
            var M = c(T, I, S, k, E + "." + I, r);
            if (M instanceof Error)
              return M;
          }
        return null;
      }
      return j(y);
    }
    function ie(c) {
      if (!Array.isArray(c))
        return process.env.NODE_ENV !== "production" && a("Invalid argument supplied to oneOfType, expected an instance of array."), i;
      for (var y = 0; y < c.length; y++) {
        var v = c[y];
        if (typeof v != "function")
          return a(
            "Invalid argument supplied to oneOfType. Expected an array of check functions, but received " + he(v) + " at index " + y + "."
          ), i;
      }
      function C(S, k, E, T, x) {
        for (var I = [], M = 0; M < c.length; M++) {
          var Y = c[M], f = Y(S, k, E, T, x, r);
          if (f == null)
            return null;
          f.data && n(f.data, "expectedType") && I.push(f.data.expectedType);
        }
        var le = I.length > 0 ? ", expected one of type [" + I.join(", ") + "]" : "";
        return new b("Invalid " + T + " `" + x + "` supplied to " + ("`" + E + "`" + le + "."));
      }
      return j(C);
    }
    function q() {
      function c(y, v, C, S, k) {
        return pe(y[v]) ? null : new b("Invalid " + S + " `" + k + "` supplied to " + ("`" + C + "`, expected a ReactNode."));
      }
      return j(c);
    }
    function we(c, y, v, C, S) {
      return new b(
        (c || "React class") + ": " + y + " type `" + v + "." + C + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + S + "`."
      );
    }
    function Le(c) {
      function y(v, C, S, k, E) {
        var T = v[C], x = oe(T);
        if (x !== "object")
          return new b("Invalid " + k + " `" + E + "` of type `" + x + "` " + ("supplied to `" + S + "`, expected `object`."));
        for (var I in c) {
          var M = c[I];
          if (typeof M != "function")
            return we(S, k, E, I, se(M));
          var Y = M(T, I, S, k, E + "." + I, r);
          if (Y)
            return Y;
        }
        return null;
      }
      return j(y);
    }
    function Se(c) {
      function y(v, C, S, k, E) {
        var T = v[C], x = oe(T);
        if (x !== "object")
          return new b("Invalid " + k + " `" + E + "` of type `" + x + "` " + ("supplied to `" + S + "`, expected `object`."));
        var I = t({}, v[C], c);
        for (var M in I) {
          var Y = c[M];
          if (n(c, M) && typeof Y != "function")
            return we(S, k, E, M, se(Y));
          if (!Y)
            return new b(
              "Invalid " + k + " `" + E + "` key `" + M + "` supplied to `" + S + "`.\nBad object: " + JSON.stringify(v[C], null, "  ") + `
Valid keys: ` + JSON.stringify(Object.keys(c), null, "  ")
            );
          var f = Y(T, M, S, k, E + "." + M, r);
          if (f)
            return f;
        }
        return null;
      }
      return j(y);
    }
    function pe(c) {
      switch (typeof c) {
        case "number":
        case "string":
        case "undefined":
          return !0;
        case "boolean":
          return !c;
        case "object":
          if (Array.isArray(c))
            return c.every(pe);
          if (c === null || s(c))
            return !0;
          var y = d(c);
          if (y) {
            var v = y.call(c), C;
            if (y !== c.entries) {
              for (; !(C = v.next()).done; )
                if (!pe(C.value))
                  return !1;
            } else
              for (; !(C = v.next()).done; ) {
                var S = C.value;
                if (S && !pe(S[1]))
                  return !1;
              }
          } else
            return !1;
          return !0;
        default:
          return !1;
      }
    }
    function Ue(c, y) {
      return c === "symbol" ? !0 : y ? y["@@toStringTag"] === "Symbol" || typeof Symbol == "function" && y instanceof Symbol : !1;
    }
    function oe(c) {
      var y = typeof c;
      return Array.isArray(c) ? "array" : c instanceof RegExp ? "object" : Ue(y, c) ? "symbol" : y;
    }
    function se(c) {
      if (typeof c > "u" || c === null)
        return "" + c;
      var y = oe(c);
      if (y === "object") {
        if (c instanceof Date)
          return "date";
        if (c instanceof RegExp)
          return "regexp";
      }
      return y;
    }
    function he(c) {
      var y = se(c);
      switch (y) {
        case "array":
        case "object":
          return "an " + y;
        case "boolean":
        case "date":
        case "regexp":
          return "a " + y;
        default:
          return y;
      }
    }
    function te(c) {
      return !c.constructor || !c.constructor.name ? m : c.constructor.name;
    }
    return g.checkPropTypes = o, g.resetWarningCache = o.resetWarningCache, g.PropTypes = g, g;
  }, Ct;
}
var At, wr;
function Oa() {
  if (wr)
    return At;
  wr = 1;
  var e = er();
  function t() {
  }
  function r() {
  }
  return r.resetWarningCache = t, At = function() {
    function n(i, s, l, h, p, d) {
      if (d !== e) {
        var m = new Error(
          "Calling PropTypes validators directly is not supported by the `prop-types` package. Use PropTypes.checkPropTypes() to call them. Read more at http://fb.me/use-check-prop-types"
        );
        throw m.name = "Invariant Violation", m;
      }
    }
    n.isRequired = n;
    function o() {
      return n;
    }
    var a = {
      array: n,
      bigint: n,
      bool: n,
      func: n,
      number: n,
      object: n,
      string: n,
      symbol: n,
      any: n,
      arrayOf: o,
      element: n,
      elementType: n,
      instanceOf: o,
      node: n,
      objectOf: o,
      oneOf: o,
      oneOfType: o,
      shape: o,
      exact: o,
      checkPropTypes: r,
      resetWarningCache: t
    };
    return a.PropTypes = a, a;
  }, At;
}
if (process.env.NODE_ENV !== "production") {
  var Ea = mn(), Ta = !0;
  Lt.exports = Sa()(Ea.isElement, Ta);
} else
  Lt.exports = Oa()();
var Ca = Lt.exports, Aa = Object.create, pt = Object.defineProperty, Ra = Object.getOwnPropertyDescriptor, xa = Object.getOwnPropertyNames, ka = Object.getPrototypeOf, Ia = Object.prototype.hasOwnProperty, Ma = (e, t) => {
  for (var r in t)
    pt(e, r, { get: t[r], enumerable: !0 });
}, vn = (e, t, r, n) => {
  if (t && typeof t == "object" || typeof t == "function")
    for (let o of xa(t))
      !Ia.call(e, o) && o !== r && pt(e, o, { get: () => t[o], enumerable: !(n = Ra(t, o)) || n.enumerable });
  return e;
}, $a = (e, t, r) => (r = e != null ? Aa(ka(e)) : {}, vn(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  t || !e || !e.__esModule ? pt(r, "default", { value: e, enumerable: !0 }) : r,
  e
)), ja = (e) => vn(pt({}, "__esModule", { value: !0 }), e), bn = {};
Ma(bn, {
  defaultProps: () => La,
  propTypes: () => Na
});
var _n = ja(bn), Da = $a(Ca);
const { string: X, bool: Z, number: Te, array: Rt, oneOfType: ze, shape: re, object: K, func: W, node: Sr } = Da.default, Na = {
  url: ze([X, Rt, K]),
  playing: Z,
  loop: Z,
  controls: Z,
  volume: Te,
  muted: Z,
  playbackRate: Te,
  width: ze([X, Te]),
  height: ze([X, Te]),
  style: K,
  progressInterval: Te,
  playsinline: Z,
  pip: Z,
  stopOnUnmount: Z,
  light: ze([Z, X, K]),
  playIcon: Sr,
  previewTabIndex: Te,
  previewAriaLabel: X,
  fallback: Sr,
  oEmbedUrl: X,
  wrapper: ze([
    X,
    W,
    re({ render: W.isRequired })
  ]),
  config: re({
    soundcloud: re({
      options: K
    }),
    youtube: re({
      playerVars: K,
      embedOptions: K,
      onUnstarted: W
    }),
    facebook: re({
      appId: X,
      version: X,
      playerId: X,
      attributes: K
    }),
    dailymotion: re({
      params: K
    }),
    vimeo: re({
      playerOptions: K,
      title: X
    }),
    mux: re({
      attributes: K,
      version: X
    }),
    file: re({
      attributes: K,
      tracks: Rt,
      forceVideo: Z,
      forceAudio: Z,
      forceHLS: Z,
      forceSafariHLS: Z,
      forceDisableHls: Z,
      forceDASH: Z,
      forceFLV: Z,
      hlsOptions: K,
      hlsVersion: X,
      dashVersion: X,
      flvVersion: X
    }),
    wistia: re({
      options: K,
      playerId: X,
      customControls: Rt
    }),
    mixcloud: re({
      options: K
    }),
    twitch: re({
      options: K,
      playerId: X
    }),
    vidyard: re({
      options: K
    })
  }),
  onReady: W,
  onStart: W,
  onPlay: W,
  onPause: W,
  onBuffer: W,
  onBufferEnd: W,
  onEnded: W,
  onError: W,
  onDuration: W,
  onSeek: W,
  onPlaybackRateChange: W,
  onPlaybackQualityChange: W,
  onProgress: W,
  onClickPreview: W,
  onEnablePIP: W,
  onDisablePIP: W
}, G = () => {
}, La = {
  playing: !1,
  loop: !1,
  controls: !1,
  volume: null,
  muted: !1,
  playbackRate: 1,
  width: "640px",
  height: "360px",
  style: {},
  progressInterval: 1e3,
  playsinline: !1,
  pip: !1,
  stopOnUnmount: !0,
  light: !1,
  fallback: null,
  wrapper: "div",
  previewTabIndex: 0,
  previewAriaLabel: "",
  oEmbedUrl: "https://noembed.com/embed?url={url}",
  config: {
    soundcloud: {
      options: {
        visual: !0,
        // Undocumented, but makes player fill container and look better
        buying: !1,
        liking: !1,
        download: !1,
        sharing: !1,
        show_comments: !1,
        show_playcount: !1
      }
    },
    youtube: {
      playerVars: {
        playsinline: 1,
        showinfo: 0,
        rel: 0,
        iv_load_policy: 3,
        modestbranding: 1
      },
      embedOptions: {},
      onUnstarted: G
    },
    facebook: {
      appId: "1309697205772819",
      version: "v3.3",
      playerId: null,
      attributes: {}
    },
    dailymotion: {
      params: {
        api: 1,
        "endscreen-enable": !1
      }
    },
    vimeo: {
      playerOptions: {
        autopause: !1,
        byline: !1,
        portrait: !1,
        title: !1
      },
      title: null
    },
    mux: {
      attributes: {},
      version: "2"
    },
    file: {
      attributes: {},
      tracks: [],
      forceVideo: !1,
      forceAudio: !1,
      forceHLS: !1,
      forceDASH: !1,
      forceFLV: !1,
      hlsOptions: {},
      hlsVersion: "1.1.4",
      dashVersion: "3.1.3",
      flvVersion: "1.5.0",
      forceDisableHls: !1
    },
    wistia: {
      options: {},
      playerId: null,
      customControls: null
    },
    mixcloud: {
      options: {
        hide_cover: 1
      }
    },
    twitch: {
      options: {},
      playerId: null
    },
    vidyard: {
      options: {}
    }
  },
  onReady: G,
  onStart: G,
  onPlay: G,
  onPause: G,
  onBuffer: G,
  onBufferEnd: G,
  onEnded: G,
  onError: G,
  onDuration: G,
  onSeek: G,
  onPlaybackRateChange: G,
  onPlaybackQualityChange: G,
  onProgress: G,
  onClickPreview: G,
  onEnablePIP: G,
  onDisablePIP: G
};
var Ua = Object.create, Be = Object.defineProperty, za = Object.getOwnPropertyDescriptor, Ha = Object.getOwnPropertyNames, Fa = Object.getPrototypeOf, Va = Object.prototype.hasOwnProperty, Ya = (e, t, r) => t in e ? Be(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r, Wa = (e, t) => {
  for (var r in t)
    Be(e, r, { get: t[r], enumerable: !0 });
}, Pn = (e, t, r, n) => {
  if (t && typeof t == "object" || typeof t == "function")
    for (let o of Ha(t))
      !Va.call(e, o) && o !== r && Be(e, o, { get: () => t[o], enumerable: !(n = za(t, o)) || n.enumerable });
  return e;
}, wn = (e, t, r) => (r = e != null ? Ua(Fa(e)) : {}, Pn(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  t || !e || !e.__esModule ? Be(r, "default", { value: e, enumerable: !0 }) : r,
  e
)), Ba = (e) => Pn(Be({}, "__esModule", { value: !0 }), e), F = (e, t, r) => (Ya(e, typeof t != "symbol" ? t + "" : t, r), r), Sn = {};
Wa(Sn, {
  default: () => dt
});
var qa = Ba(Sn), Or = wn(ye), Xa = wn(hn), On = _n, Ga = ft;
const Ka = 5e3;
class dt extends Or.Component {
  constructor() {
    super(...arguments), F(this, "mounted", !1), F(this, "isReady", !1), F(this, "isPlaying", !1), F(this, "isLoading", !0), F(this, "loadOnReady", null), F(this, "startOnPlay", !0), F(this, "seekOnPlay", null), F(this, "onDurationCalled", !1), F(this, "handlePlayerMount", (t) => {
      if (this.player) {
        this.progress();
        return;
      }
      this.player = t, this.player.load(this.props.url), this.progress();
    }), F(this, "getInternalPlayer", (t) => this.player ? this.player[t] : null), F(this, "progress", () => {
      if (this.props.url && this.player && this.isReady) {
        const t = this.getCurrentTime() || 0, r = this.getSecondsLoaded(), n = this.getDuration();
        if (n) {
          const o = {
            playedSeconds: t,
            played: t / n
          };
          r !== null && (o.loadedSeconds = r, o.loaded = r / n), (o.playedSeconds !== this.prevPlayed || o.loadedSeconds !== this.prevLoaded) && this.props.onProgress(o), this.prevPlayed = o.playedSeconds, this.prevLoaded = o.loadedSeconds;
        }
      }
      this.progressTimeout = setTimeout(this.progress, this.props.progressFrequency || this.props.progressInterval);
    }), F(this, "handleReady", () => {
      if (!this.mounted)
        return;
      this.isReady = !0, this.isLoading = !1;
      const { onReady: t, playing: r, volume: n, muted: o } = this.props;
      t(), !o && n !== null && this.player.setVolume(n), this.loadOnReady ? (this.player.load(this.loadOnReady, !0), this.loadOnReady = null) : r && this.player.play(), this.handleDurationCheck();
    }), F(this, "handlePlay", () => {
      this.isPlaying = !0, this.isLoading = !1;
      const { onStart: t, onPlay: r, playbackRate: n } = this.props;
      this.startOnPlay && (this.player.setPlaybackRate && n !== 1 && this.player.setPlaybackRate(n), t(), this.startOnPlay = !1), r(), this.seekOnPlay && (this.seekTo(this.seekOnPlay), this.seekOnPlay = null), this.handleDurationCheck();
    }), F(this, "handlePause", (t) => {
      this.isPlaying = !1, this.isLoading || this.props.onPause(t);
    }), F(this, "handleEnded", () => {
      const { activePlayer: t, loop: r, onEnded: n } = this.props;
      t.loopOnEnded && r && this.seekTo(0), r || (this.isPlaying = !1, n());
    }), F(this, "handleError", (...t) => {
      this.isLoading = !1, this.props.onError(...t);
    }), F(this, "handleDurationCheck", () => {
      clearTimeout(this.durationCheckTimeout);
      const t = this.getDuration();
      t ? this.onDurationCalled || (this.props.onDuration(t), this.onDurationCalled = !0) : this.durationCheckTimeout = setTimeout(this.handleDurationCheck, 100);
    }), F(this, "handleLoaded", () => {
      this.isLoading = !1;
    });
  }
  componentDidMount() {
    this.mounted = !0;
  }
  componentWillUnmount() {
    clearTimeout(this.progressTimeout), clearTimeout(this.durationCheckTimeout), this.isReady && this.props.stopOnUnmount && (this.player.stop(), this.player.disablePIP && this.player.disablePIP()), this.mounted = !1;
  }
  componentDidUpdate(t) {
    if (!this.player)
      return;
    const { url: r, playing: n, volume: o, muted: a, playbackRate: i, pip: s, loop: l, activePlayer: h, disableDeferredLoading: p } = this.props;
    if (!(0, Xa.default)(t.url, r)) {
      if (this.isLoading && !h.forceLoad && !p && !(0, Ga.isMediaStream)(r)) {
        console.warn(`ReactPlayer: the attempt to load ${r} is being deferred until the player has loaded`), this.loadOnReady = r;
        return;
      }
      this.isLoading = !0, this.startOnPlay = !0, this.onDurationCalled = !1, this.player.load(r, this.isReady);
    }
    !t.playing && n && !this.isPlaying && this.player.play(), t.playing && !n && this.isPlaying && this.player.pause(), !t.pip && s && this.player.enablePIP && this.player.enablePIP(), t.pip && !s && this.player.disablePIP && this.player.disablePIP(), t.volume !== o && o !== null && this.player.setVolume(o), t.muted !== a && (a ? this.player.mute() : (this.player.unmute(), o !== null && setTimeout(() => this.player.setVolume(o)))), t.playbackRate !== i && this.player.setPlaybackRate && this.player.setPlaybackRate(i), t.loop !== l && this.player.setLoop && this.player.setLoop(l);
  }
  getDuration() {
    return this.isReady ? this.player.getDuration() : null;
  }
  getCurrentTime() {
    return this.isReady ? this.player.getCurrentTime() : null;
  }
  getSecondsLoaded() {
    return this.isReady ? this.player.getSecondsLoaded() : null;
  }
  seekTo(t, r, n) {
    if (!this.isReady) {
      t !== 0 && (this.seekOnPlay = t, setTimeout(() => {
        this.seekOnPlay = null;
      }, Ka));
      return;
    }
    if (r ? r === "fraction" : t > 0 && t < 1) {
      const a = this.player.getDuration();
      if (!a) {
        console.warn("ReactPlayer: could not seek using fraction – duration not yet available");
        return;
      }
      this.player.seekTo(a * t, n);
      return;
    }
    this.player.seekTo(t, n);
  }
  render() {
    const t = this.props.activePlayer;
    return t ? /* @__PURE__ */ Or.default.createElement(
      t,
      {
        ...this.props,
        onMount: this.handlePlayerMount,
        onReady: this.handleReady,
        onPlay: this.handlePlay,
        onPause: this.handlePause,
        onEnded: this.handleEnded,
        onLoaded: this.handleLoaded,
        onError: this.handleError
      }
    ) : null;
  }
}
F(dt, "displayName", "Player");
F(dt, "propTypes", On.propTypes);
F(dt, "defaultProps", On.defaultProps);
var Ja = Object.create, qe = Object.defineProperty, Za = Object.getOwnPropertyDescriptor, Qa = Object.getOwnPropertyNames, ei = Object.getPrototypeOf, ti = Object.prototype.hasOwnProperty, ri = (e, t, r) => t in e ? qe(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r, ni = (e, t) => {
  for (var r in t)
    qe(e, r, { get: t[r], enumerable: !0 });
}, En = (e, t, r, n) => {
  if (t && typeof t == "object" || typeof t == "function")
    for (let o of Qa(t))
      !ti.call(e, o) && o !== r && qe(e, o, { get: () => t[o], enumerable: !(n = Za(t, o)) || n.enumerable });
  return e;
}, Xe = (e, t, r) => (r = e != null ? Ja(ei(e)) : {}, En(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  t || !e || !e.__esModule ? qe(r, "default", { value: e, enumerable: !0 }) : r,
  e
)), oi = (e) => En(qe({}, "__esModule", { value: !0 }), e), H = (e, t, r) => (ri(e, typeof t != "symbol" ? t + "" : t, r), r), Tn = {};
ni(Tn, {
  createReactPlayer: () => di
});
var ai = oi(Tn), Ae = Xe(ye), ii = Xe(qr), xt = Xe(ya), Er = Xe(hn), Fe = _n, Cn = ft, si = Xe(qa);
const ci = (0, Cn.lazy)(() => import(
  /* webpackChunkName: 'reactPlayerPreview' */
  "./Preview-CaGUgWVm.js"
).then((e) => e.P)), li = typeof window < "u" && window.document && typeof document < "u", ui = typeof Pt < "u" && Pt.window && Pt.window.document, fi = Object.keys(Fe.propTypes), pi = li || ui ? Ae.Suspense : () => null, He = [], di = (e, t) => {
  var r;
  return r = class extends Ae.Component {
    constructor() {
      super(...arguments), H(this, "state", {
        showPreview: !!this.props.light
      }), H(this, "references", {
        wrapper: (n) => {
          this.wrapper = n;
        },
        player: (n) => {
          this.player = n;
        }
      }), H(this, "handleClickPreview", (n) => {
        this.setState({ showPreview: !1 }), this.props.onClickPreview(n);
      }), H(this, "showPreview", () => {
        this.setState({ showPreview: !0 });
      }), H(this, "getDuration", () => this.player ? this.player.getDuration() : null), H(this, "getCurrentTime", () => this.player ? this.player.getCurrentTime() : null), H(this, "getSecondsLoaded", () => this.player ? this.player.getSecondsLoaded() : null), H(this, "getInternalPlayer", (n = "player") => this.player ? this.player.getInternalPlayer(n) : null), H(this, "seekTo", (n, o, a) => {
        if (!this.player)
          return null;
        this.player.seekTo(n, o, a);
      }), H(this, "handleReady", () => {
        this.props.onReady(this);
      }), H(this, "getActivePlayer", (0, xt.default)((n) => {
        for (const o of [...He, ...e])
          if (o.canPlay(n))
            return o;
        return t || null;
      })), H(this, "getConfig", (0, xt.default)((n, o) => {
        const { config: a } = this.props;
        return ii.default.all([
          Fe.defaultProps.config,
          Fe.defaultProps.config[o] || {},
          a,
          a[o] || {}
        ]);
      })), H(this, "getAttributes", (0, xt.default)((n) => (0, Cn.omit)(this.props, fi))), H(this, "renderActivePlayer", (n) => {
        if (!n)
          return null;
        const o = this.getActivePlayer(n);
        if (!o)
          return null;
        const a = this.getConfig(n, o.key);
        return /* @__PURE__ */ Ae.default.createElement(
          si.default,
          {
            ...this.props,
            key: o.key,
            ref: this.references.player,
            config: a,
            activePlayer: o.lazyPlayer || o,
            onReady: this.handleReady
          }
        );
      });
    }
    shouldComponentUpdate(n, o) {
      return !(0, Er.default)(this.props, n) || !(0, Er.default)(this.state, o);
    }
    componentDidUpdate(n) {
      const { light: o } = this.props;
      !n.light && o && this.setState({ showPreview: !0 }), n.light && !o && this.setState({ showPreview: !1 });
    }
    renderPreview(n) {
      if (!n)
        return null;
      const { light: o, playIcon: a, previewTabIndex: i, oEmbedUrl: s, previewAriaLabel: l } = this.props;
      return /* @__PURE__ */ Ae.default.createElement(
        ci,
        {
          url: n,
          light: o,
          playIcon: a,
          previewTabIndex: i,
          previewAriaLabel: l,
          oEmbedUrl: s,
          onClick: this.handleClickPreview
        }
      );
    }
    render() {
      const { url: n, style: o, width: a, height: i, fallback: s, wrapper: l } = this.props, { showPreview: h } = this.state, p = this.getAttributes(n), d = typeof l == "string" ? this.references.wrapper : void 0;
      return /* @__PURE__ */ Ae.default.createElement(l, { ref: d, style: { ...o, width: a, height: i }, ...p }, /* @__PURE__ */ Ae.default.createElement(pi, { fallback: s }, h ? this.renderPreview(n) : this.renderActivePlayer(n)));
    }
  }, H(r, "displayName", "ReactPlayer"), H(r, "propTypes", Fe.propTypes), H(r, "defaultProps", Fe.defaultProps), H(r, "addCustomPlayer", (n) => {
    He.push(n);
  }), H(r, "removeCustomPlayers", () => {
    He.length = 0;
  }), H(r, "canPlay", (n) => {
    for (const o of [...He, ...e])
      if (o.canPlay(n))
        return !0;
    return !1;
  }), H(r, "canEnablePIP", (n) => {
    for (const o of [...He, ...e])
      if (o.canEnablePIP && o.canEnablePIP(n))
        return !0;
    return !1;
  }), r;
};
var yi = Object.create, yt = Object.defineProperty, hi = Object.getOwnPropertyDescriptor, mi = Object.getOwnPropertyNames, gi = Object.getPrototypeOf, vi = Object.prototype.hasOwnProperty, bi = (e, t) => {
  for (var r in t)
    yt(e, r, { get: t[r], enumerable: !0 });
}, An = (e, t, r, n) => {
  if (t && typeof t == "object" || typeof t == "function")
    for (let o of mi(t))
      !vi.call(e, o) && o !== r && yt(e, o, { get: () => t[o], enumerable: !(n = hi(t, o)) || n.enumerable });
  return e;
}, _i = (e, t, r) => (r = e != null ? yi(gi(e)) : {}, An(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  t || !e || !e.__esModule ? yt(r, "default", { value: e, enumerable: !0 }) : r,
  e
)), Pi = (e) => An(yt({}, "__esModule", { value: !0 }), e), Rn = {};
bi(Rn, {
  default: () => Ei
});
var wi = Pi(Rn), Ut = _i(ca), Si = ai;
const Oi = Ut.default[Ut.default.length - 1];
var Ei = (0, Si.createReactPlayer)(Ut.default, Oi);
const Tr = /* @__PURE__ */ no(wi);
var ee = function() {
  return ee = Object.assign || function(t) {
    for (var r, n = 1, o = arguments.length; n < o; n++) {
      r = arguments[n];
      for (var a in r)
        Object.prototype.hasOwnProperty.call(r, a) && (t[a] = r[a]);
    }
    return t;
  }, ee.apply(this, arguments);
};
function xe(e, t, r) {
  if (r || arguments.length === 2)
    for (var n = 0, o = t.length, a; n < o; n++)
      (a || !(n in t)) && (a || (a = Array.prototype.slice.call(t, 0, n)), a[n] = t[n]);
  return e.concat(a || Array.prototype.slice.call(t));
}
function Ti(e) {
  var t = /* @__PURE__ */ Object.create(null);
  return function(r) {
    return t[r] === void 0 && (t[r] = e(r)), t[r];
  };
}
var Ci = /^((children|dangerouslySetInnerHTML|key|ref|autoFocus|defaultValue|defaultChecked|innerHTML|suppressContentEditableWarning|suppressHydrationWarning|valueLink|abbr|accept|acceptCharset|accessKey|action|allow|allowUserMedia|allowPaymentRequest|allowFullScreen|allowTransparency|alt|async|autoComplete|autoPlay|capture|cellPadding|cellSpacing|challenge|charSet|checked|cite|classID|className|cols|colSpan|content|contentEditable|contextMenu|controls|controlsList|coords|crossOrigin|data|dateTime|decoding|default|defer|dir|disabled|disablePictureInPicture|disableRemotePlayback|download|draggable|encType|enterKeyHint|form|formAction|formEncType|formMethod|formNoValidate|formTarget|frameBorder|headers|height|hidden|high|href|hrefLang|htmlFor|httpEquiv|id|inputMode|integrity|is|keyParams|keyType|kind|label|lang|list|loading|loop|low|marginHeight|marginWidth|max|maxLength|media|mediaGroup|method|min|minLength|multiple|muted|name|nonce|noValidate|open|optimum|pattern|placeholder|playsInline|poster|preload|profile|radioGroup|readOnly|referrerPolicy|rel|required|reversed|role|rows|rowSpan|sandbox|scope|scoped|scrolling|seamless|selected|shape|size|sizes|slot|span|spellCheck|src|srcDoc|srcLang|srcSet|start|step|style|summary|tabIndex|target|title|translate|type|useMap|value|width|wmode|wrap|about|datatype|inlist|prefix|property|resource|typeof|vocab|autoCapitalize|autoCorrect|autoSave|color|incremental|fallback|inert|itemProp|itemScope|itemType|itemID|itemRef|on|option|results|security|unselectable|accentHeight|accumulate|additive|alignmentBaseline|allowReorder|alphabetic|amplitude|arabicForm|ascent|attributeName|attributeType|autoReverse|azimuth|baseFrequency|baselineShift|baseProfile|bbox|begin|bias|by|calcMode|capHeight|clip|clipPathUnits|clipPath|clipRule|colorInterpolation|colorInterpolationFilters|colorProfile|colorRendering|contentScriptType|contentStyleType|cursor|cx|cy|d|decelerate|descent|diffuseConstant|direction|display|divisor|dominantBaseline|dur|dx|dy|edgeMode|elevation|enableBackground|end|exponent|externalResourcesRequired|fill|fillOpacity|fillRule|filter|filterRes|filterUnits|floodColor|floodOpacity|focusable|fontFamily|fontSize|fontSizeAdjust|fontStretch|fontStyle|fontVariant|fontWeight|format|from|fr|fx|fy|g1|g2|glyphName|glyphOrientationHorizontal|glyphOrientationVertical|glyphRef|gradientTransform|gradientUnits|hanging|horizAdvX|horizOriginX|ideographic|imageRendering|in|in2|intercept|k|k1|k2|k3|k4|kernelMatrix|kernelUnitLength|kerning|keyPoints|keySplines|keyTimes|lengthAdjust|letterSpacing|lightingColor|limitingConeAngle|local|markerEnd|markerMid|markerStart|markerHeight|markerUnits|markerWidth|mask|maskContentUnits|maskUnits|mathematical|mode|numOctaves|offset|opacity|operator|order|orient|orientation|origin|overflow|overlinePosition|overlineThickness|panose1|paintOrder|pathLength|patternContentUnits|patternTransform|patternUnits|pointerEvents|points|pointsAtX|pointsAtY|pointsAtZ|preserveAlpha|preserveAspectRatio|primitiveUnits|r|radius|refX|refY|renderingIntent|repeatCount|repeatDur|requiredExtensions|requiredFeatures|restart|result|rotate|rx|ry|scale|seed|shapeRendering|slope|spacing|specularConstant|specularExponent|speed|spreadMethod|startOffset|stdDeviation|stemh|stemv|stitchTiles|stopColor|stopOpacity|strikethroughPosition|strikethroughThickness|string|stroke|strokeDasharray|strokeDashoffset|strokeLinecap|strokeLinejoin|strokeMiterlimit|strokeOpacity|strokeWidth|surfaceScale|systemLanguage|tableValues|targetX|targetY|textAnchor|textDecoration|textRendering|textLength|to|transform|u1|u2|underlinePosition|underlineThickness|unicode|unicodeBidi|unicodeRange|unitsPerEm|vAlphabetic|vHanging|vIdeographic|vMathematical|values|vectorEffect|version|vertAdvY|vertOriginX|vertOriginY|viewBox|viewTarget|visibility|widths|wordSpacing|writingMode|x|xHeight|x1|x2|xChannelSelector|xlinkActuate|xlinkArcrole|xlinkHref|xlinkRole|xlinkShow|xlinkTitle|xlinkType|xmlBase|xmlns|xmlnsXlink|xmlLang|xmlSpace|y|y1|y2|yChannelSelector|z|zoomAndPan|for|class|autofocus)|(([Dd][Aa][Tt][Aa]|[Aa][Rr][Ii][Aa]|x)-.*))$/, Ai = /* @__PURE__ */ Ti(
  function(e) {
    return Ci.test(e) || e.charCodeAt(0) === 111 && e.charCodeAt(1) === 110 && e.charCodeAt(2) < 91;
  }
  /* Z+1 */
), z = "-ms-", Ye = "-moz-", U = "-webkit-", xn = "comm", ht = "rule", tr = "decl", Ri = "@import", kn = "@keyframes", xi = "@layer", In = Math.abs, rr = String.fromCharCode, zt = Object.assign;
function ki(e, t) {
  return B(e, 0) ^ 45 ? (((t << 2 ^ B(e, 0)) << 2 ^ B(e, 1)) << 2 ^ B(e, 2)) << 2 ^ B(e, 3) : 0;
}
function Mn(e) {
  return e.trim();
}
function fe(e, t) {
  return (e = t.exec(e)) ? e[0] : e;
}
function A(e, t, r) {
  return e.replace(t, r);
}
function rt(e, t, r) {
  return e.indexOf(t, r);
}
function B(e, t) {
  return e.charCodeAt(t) | 0;
}
function ke(e, t, r) {
  return e.slice(t, r);
}
function ue(e) {
  return e.length;
}
function $n(e) {
  return e.length;
}
function Ve(e, t) {
  return t.push(e), e;
}
function Ii(e, t) {
  return e.map(t).join("");
}
function Cr(e, t) {
  return e.filter(function(r) {
    return !fe(r, t);
  });
}
var mt = 1, Ie = 1, jn = 0, ae = 0, V = 0, De = "";
function gt(e, t, r, n, o, a, i, s) {
  return { value: e, root: t, parent: r, type: n, props: o, children: a, line: mt, column: Ie, length: i, return: "", siblings: s };
}
function de(e, t) {
  return zt(gt("", null, null, "", null, null, 0, e.siblings), e, { length: -e.length }, t);
}
function Ce(e) {
  for (; e.root; )
    e = de(e.root, { children: [e] });
  Ve(e, e.siblings);
}
function Mi() {
  return V;
}
function $i() {
  return V = ae > 0 ? B(De, --ae) : 0, Ie--, V === 10 && (Ie = 1, mt--), V;
}
function ce() {
  return V = ae < jn ? B(De, ae++) : 0, Ie++, V === 10 && (Ie = 1, mt++), V;
}
function be() {
  return B(De, ae);
}
function nt() {
  return ae;
}
function vt(e, t) {
  return ke(De, e, t);
}
function Ht(e) {
  switch (e) {
    case 0:
    case 9:
    case 10:
    case 13:
    case 32:
      return 5;
    case 33:
    case 43:
    case 44:
    case 47:
    case 62:
    case 64:
    case 126:
    case 59:
    case 123:
    case 125:
      return 4;
    case 58:
      return 3;
    case 34:
    case 39:
    case 40:
    case 91:
      return 2;
    case 41:
    case 93:
      return 1;
  }
  return 0;
}
function ji(e) {
  return mt = Ie = 1, jn = ue(De = e), ae = 0, [];
}
function Di(e) {
  return De = "", e;
}
function kt(e) {
  return Mn(vt(ae - 1, Ft(e === 91 ? e + 2 : e === 40 ? e + 1 : e)));
}
function Ni(e) {
  for (; (V = be()) && V < 33; )
    ce();
  return Ht(e) > 2 || Ht(V) > 3 ? "" : " ";
}
function Li(e, t) {
  for (; --t && ce() && !(V < 48 || V > 102 || V > 57 && V < 65 || V > 70 && V < 97); )
    ;
  return vt(e, nt() + (t < 6 && be() == 32 && ce() == 32));
}
function Ft(e) {
  for (; ce(); )
    switch (V) {
      case e:
        return ae;
      case 34:
      case 39:
        e !== 34 && e !== 39 && Ft(V);
        break;
      case 40:
        e === 41 && Ft(e);
        break;
      case 92:
        ce();
        break;
    }
  return ae;
}
function Ui(e, t) {
  for (; ce() && e + V !== 57; )
    if (e + V === 84 && be() === 47)
      break;
  return "/*" + vt(t, ae - 1) + "*" + rr(e === 47 ? e : ce());
}
function zi(e) {
  for (; !Ht(be()); )
    ce();
  return vt(e, ae);
}
function Hi(e) {
  return Di(ot("", null, null, null, [""], e = ji(e), 0, [0], e));
}
function ot(e, t, r, n, o, a, i, s, l) {
  for (var h = 0, p = 0, d = i, m = 0, g = 0, R = 0, b = 1, j = 1, $ = 1, D = 0, _ = "", P = o, u = a, O = n, w = _; j; )
    switch (R = D, D = ce()) {
      case 40:
        if (R != 108 && B(w, d - 1) == 58) {
          rt(w += A(kt(D), "&", "&\f"), "&\f", In(h ? s[h - 1] : 0)) != -1 && ($ = -1);
          break;
        }
      case 34:
      case 39:
      case 91:
        w += kt(D);
        break;
      case 9:
      case 10:
      case 13:
      case 32:
        w += Ni(R);
        break;
      case 92:
        w += Li(nt() - 1, 7);
        continue;
      case 47:
        switch (be()) {
          case 42:
          case 47:
            Ve(Fi(Ui(ce(), nt()), t, r, l), l);
            break;
          default:
            w += "/";
        }
        break;
      case 123 * b:
        s[h++] = ue(w) * $;
      case 125 * b:
      case 59:
      case 0:
        switch (D) {
          case 0:
          case 125:
            j = 0;
          case 59 + p:
            $ == -1 && (w = A(w, /\f/g, "")), g > 0 && ue(w) - d && Ve(g > 32 ? Rr(w + ";", n, r, d - 1, l) : Rr(A(w, " ", "") + ";", n, r, d - 2, l), l);
            break;
          case 59:
            w += ";";
          default:
            if (Ve(O = Ar(w, t, r, h, p, o, s, _, P = [], u = [], d, a), a), D === 123)
              if (p === 0)
                ot(w, t, O, O, P, a, d, s, u);
              else
                switch (m === 99 && B(w, 3) === 110 ? 100 : m) {
                  case 100:
                  case 108:
                  case 109:
                  case 115:
                    ot(e, O, O, n && Ve(Ar(e, O, O, 0, 0, o, s, _, o, P = [], d, u), u), o, u, d, s, n ? P : u);
                    break;
                  default:
                    ot(w, O, O, O, [""], u, 0, s, u);
                }
        }
        h = p = g = 0, b = $ = 1, _ = w = "", d = i;
        break;
      case 58:
        d = 1 + ue(w), g = R;
      default:
        if (b < 1) {
          if (D == 123)
            --b;
          else if (D == 125 && b++ == 0 && $i() == 125)
            continue;
        }
        switch (w += rr(D), D * b) {
          case 38:
            $ = p > 0 ? 1 : (w += "\f", -1);
            break;
          case 44:
            s[h++] = (ue(w) - 1) * $, $ = 1;
            break;
          case 64:
            be() === 45 && (w += kt(ce())), m = be(), p = d = ue(_ = w += zi(nt())), D++;
            break;
          case 45:
            R === 45 && ue(w) == 2 && (b = 0);
        }
    }
  return a;
}
function Ar(e, t, r, n, o, a, i, s, l, h, p, d) {
  for (var m = o - 1, g = o === 0 ? a : [""], R = $n(g), b = 0, j = 0, $ = 0; b < n; ++b)
    for (var D = 0, _ = ke(e, m + 1, m = In(j = i[b])), P = e; D < R; ++D)
      (P = Mn(j > 0 ? g[D] + " " + _ : A(_, /&\f/g, g[D]))) && (l[$++] = P);
  return gt(e, t, r, o === 0 ? ht : s, l, h, p, d);
}
function Fi(e, t, r, n) {
  return gt(e, t, r, xn, rr(Mi()), ke(e, 2, -2), 0, n);
}
function Rr(e, t, r, n, o) {
  return gt(e, t, r, tr, ke(e, 0, n), ke(e, n + 1, -1), n, o);
}
function Dn(e, t, r) {
  switch (ki(e, t)) {
    case 5103:
      return U + "print-" + e + e;
    case 5737:
    case 4201:
    case 3177:
    case 3433:
    case 1641:
    case 4457:
    case 2921:
    case 5572:
    case 6356:
    case 5844:
    case 3191:
    case 6645:
    case 3005:
    case 6391:
    case 5879:
    case 5623:
    case 6135:
    case 4599:
    case 4855:
    case 4215:
    case 6389:
    case 5109:
    case 5365:
    case 5621:
    case 3829:
      return U + e + e;
    case 4789:
      return Ye + e + e;
    case 5349:
    case 4246:
    case 4810:
    case 6968:
    case 2756:
      return U + e + Ye + e + z + e + e;
    case 5936:
      switch (B(e, t + 11)) {
        case 114:
          return U + e + z + A(e, /[svh]\w+-[tblr]{2}/, "tb") + e;
        case 108:
          return U + e + z + A(e, /[svh]\w+-[tblr]{2}/, "tb-rl") + e;
        case 45:
          return U + e + z + A(e, /[svh]\w+-[tblr]{2}/, "lr") + e;
      }
    case 6828:
    case 4268:
    case 2903:
      return U + e + z + e + e;
    case 6165:
      return U + e + z + "flex-" + e + e;
    case 5187:
      return U + e + A(e, /(\w+).+(:[^]+)/, U + "box-$1$2" + z + "flex-$1$2") + e;
    case 5443:
      return U + e + z + "flex-item-" + A(e, /flex-|-self/g, "") + (fe(e, /flex-|baseline/) ? "" : z + "grid-row-" + A(e, /flex-|-self/g, "")) + e;
    case 4675:
      return U + e + z + "flex-line-pack" + A(e, /align-content|flex-|-self/g, "") + e;
    case 5548:
      return U + e + z + A(e, "shrink", "negative") + e;
    case 5292:
      return U + e + z + A(e, "basis", "preferred-size") + e;
    case 6060:
      return U + "box-" + A(e, "-grow", "") + U + e + z + A(e, "grow", "positive") + e;
    case 4554:
      return U + A(e, /([^-])(transform)/g, "$1" + U + "$2") + e;
    case 6187:
      return A(A(A(e, /(zoom-|grab)/, U + "$1"), /(image-set)/, U + "$1"), e, "") + e;
    case 5495:
    case 3959:
      return A(e, /(image-set\([^]*)/, U + "$1$`$1");
    case 4968:
      return A(A(e, /(.+:)(flex-)?(.*)/, U + "box-pack:$3" + z + "flex-pack:$3"), /s.+-b[^;]+/, "justify") + U + e + e;
    case 4200:
      if (!fe(e, /flex-|baseline/))
        return z + "grid-column-align" + ke(e, t) + e;
      break;
    case 2592:
    case 3360:
      return z + A(e, "template-", "") + e;
    case 4384:
    case 3616:
      return r && r.some(function(n, o) {
        return t = o, fe(n.props, /grid-\w+-end/);
      }) ? ~rt(e + (r = r[t].value), "span", 0) ? e : z + A(e, "-start", "") + e + z + "grid-row-span:" + (~rt(r, "span", 0) ? fe(r, /\d+/) : +fe(r, /\d+/) - +fe(e, /\d+/)) + ";" : z + A(e, "-start", "") + e;
    case 4896:
    case 4128:
      return r && r.some(function(n) {
        return fe(n.props, /grid-\w+-start/);
      }) ? e : z + A(A(e, "-end", "-span"), "span ", "") + e;
    case 4095:
    case 3583:
    case 4068:
    case 2532:
      return A(e, /(.+)-inline(.+)/, U + "$1$2") + e;
    case 8116:
    case 7059:
    case 5753:
    case 5535:
    case 5445:
    case 5701:
    case 4933:
    case 4677:
    case 5533:
    case 5789:
    case 5021:
    case 4765:
      if (ue(e) - 1 - t > 6)
        switch (B(e, t + 1)) {
          case 109:
            if (B(e, t + 4) !== 45)
              break;
          case 102:
            return A(e, /(.+:)(.+)-([^]+)/, "$1" + U + "$2-$3$1" + Ye + (B(e, t + 3) == 108 ? "$3" : "$2-$3")) + e;
          case 115:
            return ~rt(e, "stretch", 0) ? Dn(A(e, "stretch", "fill-available"), t, r) + e : e;
        }
      break;
    case 5152:
    case 5920:
      return A(e, /(.+?):(\d+)(\s*\/\s*(span)?\s*(\d+))?(.*)/, function(n, o, a, i, s, l, h) {
        return z + o + ":" + a + h + (i ? z + o + "-span:" + (s ? l : +l - +a) + h : "") + e;
      });
    case 4949:
      if (B(e, t + 6) === 121)
        return A(e, ":", ":" + U) + e;
      break;
    case 6444:
      switch (B(e, B(e, 14) === 45 ? 18 : 11)) {
        case 120:
          return A(e, /(.+:)([^;\s!]+)(;|(\s+)?!.+)?/, "$1" + U + (B(e, 14) === 45 ? "inline-" : "") + "box$3$1" + U + "$2$3$1" + z + "$2box$3") + e;
        case 100:
          return A(e, ":", ":" + z) + e;
      }
      break;
    case 5719:
    case 2647:
    case 2135:
    case 3927:
    case 2391:
      return A(e, "scroll-", "scroll-snap-") + e;
  }
  return e;
}
function st(e, t) {
  for (var r = "", n = 0; n < e.length; n++)
    r += t(e[n], n, e, t) || "";
  return r;
}
function Vi(e, t, r, n) {
  switch (e.type) {
    case xi:
      if (e.children.length)
        break;
    case Ri:
    case tr:
      return e.return = e.return || e.value;
    case xn:
      return "";
    case kn:
      return e.return = e.value + "{" + st(e.children, n) + "}";
    case ht:
      if (!ue(e.value = e.props.join(",")))
        return "";
  }
  return ue(r = st(e.children, n)) ? e.return = e.value + "{" + r + "}" : "";
}
function Yi(e) {
  var t = $n(e);
  return function(r, n, o, a) {
    for (var i = "", s = 0; s < t; s++)
      i += e[s](r, n, o, a) || "";
    return i;
  };
}
function Wi(e) {
  return function(t) {
    t.root || (t = t.return) && e(t);
  };
}
function Bi(e, t, r, n) {
  if (e.length > -1 && !e.return)
    switch (e.type) {
      case tr:
        e.return = Dn(e.value, e.length, r);
        return;
      case kn:
        return st([de(e, { value: A(e.value, "@", "@" + U) })], n);
      case ht:
        if (e.length)
          return Ii(r = e.props, function(o) {
            switch (fe(o, n = /(::plac\w+|:read-\w+)/)) {
              case ":read-only":
              case ":read-write":
                Ce(de(e, { props: [A(o, /:(read-\w+)/, ":" + Ye + "$1")] })), Ce(de(e, { props: [o] })), zt(e, { props: Cr(r, n) });
                break;
              case "::placeholder":
                Ce(de(e, { props: [A(o, /:(plac\w+)/, ":" + U + "input-$1")] })), Ce(de(e, { props: [A(o, /:(plac\w+)/, ":" + Ye + "$1")] })), Ce(de(e, { props: [A(o, /:(plac\w+)/, z + "input-$1")] })), Ce(de(e, { props: [o] })), zt(e, { props: Cr(r, n) });
                break;
            }
            return "";
          });
    }
}
var qi = {
  animationIterationCount: 1,
  aspectRatio: 1,
  borderImageOutset: 1,
  borderImageSlice: 1,
  borderImageWidth: 1,
  boxFlex: 1,
  boxFlexGroup: 1,
  boxOrdinalGroup: 1,
  columnCount: 1,
  columns: 1,
  flex: 1,
  flexGrow: 1,
  flexPositive: 1,
  flexShrink: 1,
  flexNegative: 1,
  flexOrder: 1,
  gridRow: 1,
  gridRowEnd: 1,
  gridRowSpan: 1,
  gridRowStart: 1,
  gridColumn: 1,
  gridColumnEnd: 1,
  gridColumnSpan: 1,
  gridColumnStart: 1,
  msGridRow: 1,
  msGridRowSpan: 1,
  msGridColumn: 1,
  msGridColumnSpan: 1,
  fontWeight: 1,
  lineHeight: 1,
  opacity: 1,
  order: 1,
  orphans: 1,
  tabSize: 1,
  widows: 1,
  zIndex: 1,
  zoom: 1,
  WebkitLineClamp: 1,
  // SVG-related properties
  fillOpacity: 1,
  floodOpacity: 1,
  stopOpacity: 1,
  strokeDasharray: 1,
  strokeDashoffset: 1,
  strokeMiterlimit: 1,
  strokeOpacity: 1,
  strokeWidth: 1
}, Pe = typeof process < "u" && process.env !== void 0 && (process.env.REACT_APP_SC_ATTR || process.env.SC_ATTR) || "data-styled", Nn = "active", Ln = "data-styled-version", bt = "6.1.13", nr = `/*!sc*/
`, ct = typeof window < "u" && "HTMLElement" in window, Xi = !!(typeof SC_DISABLE_SPEEDY == "boolean" ? SC_DISABLE_SPEEDY : typeof process < "u" && process.env !== void 0 && process.env.REACT_APP_SC_DISABLE_SPEEDY !== void 0 && process.env.REACT_APP_SC_DISABLE_SPEEDY !== "" ? process.env.REACT_APP_SC_DISABLE_SPEEDY !== "false" && process.env.REACT_APP_SC_DISABLE_SPEEDY : typeof process < "u" && process.env !== void 0 && process.env.SC_DISABLE_SPEEDY !== void 0 && process.env.SC_DISABLE_SPEEDY !== "" ? process.env.SC_DISABLE_SPEEDY !== "false" && process.env.SC_DISABLE_SPEEDY : process.env.NODE_ENV !== "production"), xr = /invalid hook call/i, Je = /* @__PURE__ */ new Set(), Gi = function(e, t) {
  if (process.env.NODE_ENV !== "production") {
    var r = t ? ' with the id of "'.concat(t, '"') : "", n = "The component ".concat(e).concat(r, ` has been created dynamically.
`) + `You may see this warning because you've called styled inside another component.
To resolve this only create new StyledComponents outside of any render method and function component.`, o = console.error;
    try {
      var a = !0;
      console.error = function(i) {
        for (var s = [], l = 1; l < arguments.length; l++)
          s[l - 1] = arguments[l];
        xr.test(i) ? (a = !1, Je.delete(n)) : o.apply(void 0, xe([i], s, !1));
      }, Wr(), a && !Je.has(n) && (console.warn(n), Je.add(n));
    } catch (i) {
      xr.test(i.message) && Je.delete(n);
    } finally {
      console.error = o;
    }
  }
}, _t = Object.freeze([]), Me = Object.freeze({});
function Ki(e, t, r) {
  return r === void 0 && (r = Me), e.theme !== r.theme && e.theme || t || r.theme;
}
var Vt = /* @__PURE__ */ new Set(["a", "abbr", "address", "area", "article", "aside", "audio", "b", "base", "bdi", "bdo", "big", "blockquote", "body", "br", "button", "canvas", "caption", "cite", "code", "col", "colgroup", "data", "datalist", "dd", "del", "details", "dfn", "dialog", "div", "dl", "dt", "em", "embed", "fieldset", "figcaption", "figure", "footer", "form", "h1", "h2", "h3", "h4", "h5", "h6", "header", "hgroup", "hr", "html", "i", "iframe", "img", "input", "ins", "kbd", "keygen", "label", "legend", "li", "link", "main", "map", "mark", "menu", "menuitem", "meta", "meter", "nav", "noscript", "object", "ol", "optgroup", "option", "output", "p", "param", "picture", "pre", "progress", "q", "rp", "rt", "ruby", "s", "samp", "script", "section", "select", "small", "source", "span", "strong", "style", "sub", "summary", "sup", "table", "tbody", "td", "textarea", "tfoot", "th", "thead", "time", "tr", "track", "u", "ul", "use", "var", "video", "wbr", "circle", "clipPath", "defs", "ellipse", "foreignObject", "g", "image", "line", "linearGradient", "marker", "mask", "path", "pattern", "polygon", "polyline", "radialGradient", "rect", "stop", "svg", "text", "tspan"]), Ji = /[!"#$%&'()*+,./:;<=>?@[\\\]^`{|}~-]+/g, Zi = /(^-|-$)/g;
function kr(e) {
  return e.replace(Ji, "-").replace(Zi, "");
}
var Qi = /(a)(d)/gi, Ze = 52, Ir = function(e) {
  return String.fromCharCode(e + (e > 25 ? 39 : 97));
};
function Yt(e) {
  var t, r = "";
  for (t = Math.abs(e); t > Ze; t = t / Ze | 0)
    r = Ir(t % Ze) + r;
  return (Ir(t % Ze) + r).replace(Qi, "$1-$2");
}
var It, Un = 5381, ge = function(e, t) {
  for (var r = t.length; r; )
    e = 33 * e ^ t.charCodeAt(--r);
  return e;
}, zn = function(e) {
  return ge(Un, e);
};
function es(e) {
  return Yt(zn(e) >>> 0);
}
function Hn(e) {
  return process.env.NODE_ENV !== "production" && typeof e == "string" && e || e.displayName || e.name || "Component";
}
function Mt(e) {
  return typeof e == "string" && (process.env.NODE_ENV === "production" || e.charAt(0) === e.charAt(0).toLowerCase());
}
var Fn = typeof Symbol == "function" && Symbol.for, Vn = Fn ? Symbol.for("react.memo") : 60115, ts = Fn ? Symbol.for("react.forward_ref") : 60112, rs = { childContextTypes: !0, contextType: !0, contextTypes: !0, defaultProps: !0, displayName: !0, getDefaultProps: !0, getDerivedStateFromError: !0, getDerivedStateFromProps: !0, mixins: !0, propTypes: !0, type: !0 }, ns = { name: !0, length: !0, prototype: !0, caller: !0, callee: !0, arguments: !0, arity: !0 }, Yn = { $$typeof: !0, compare: !0, defaultProps: !0, displayName: !0, propTypes: !0, type: !0 }, os = ((It = {})[ts] = { $$typeof: !0, render: !0, defaultProps: !0, displayName: !0, propTypes: !0 }, It[Vn] = Yn, It);
function Mr(e) {
  return ("type" in (t = e) && t.type.$$typeof) === Vn ? Yn : "$$typeof" in e ? os[e.$$typeof] : rs;
  var t;
}
var as = Object.defineProperty, is = Object.getOwnPropertyNames, $r = Object.getOwnPropertySymbols, ss = Object.getOwnPropertyDescriptor, cs = Object.getPrototypeOf, jr = Object.prototype;
function Wn(e, t, r) {
  if (typeof t != "string") {
    if (jr) {
      var n = cs(t);
      n && n !== jr && Wn(e, n, r);
    }
    var o = is(t);
    $r && (o = o.concat($r(t)));
    for (var a = Mr(e), i = Mr(t), s = 0; s < o.length; ++s) {
      var l = o[s];
      if (!(l in ns || r && r[l] || i && l in i || a && l in a)) {
        var h = ss(t, l);
        try {
          as(e, l, h);
        } catch {
        }
      }
    }
  }
  return e;
}
function $e(e) {
  return typeof e == "function";
}
function or(e) {
  return typeof e == "object" && "styledComponentId" in e;
}
function ve(e, t) {
  return e && t ? "".concat(e, " ").concat(t) : e || t || "";
}
function Dr(e, t) {
  if (e.length === 0)
    return "";
  for (var r = e[0], n = 1; n < e.length; n++)
    r += t ? t + e[n] : e[n];
  return r;
}
function je(e) {
  return e !== null && typeof e == "object" && e.constructor.name === Object.name && !("props" in e && e.$$typeof);
}
function Wt(e, t, r) {
  if (r === void 0 && (r = !1), !r && !je(e) && !Array.isArray(e))
    return t;
  if (Array.isArray(t))
    for (var n = 0; n < t.length; n++)
      e[n] = Wt(e[n], t[n]);
  else if (je(t))
    for (var n in t)
      e[n] = Wt(e[n], t[n]);
  return e;
}
function ar(e, t) {
  Object.defineProperty(e, "toString", { value: t });
}
var ls = process.env.NODE_ENV !== "production" ? { 1: `Cannot create styled-component for component: %s.

`, 2: `Can't collect styles once you've consumed a \`ServerStyleSheet\`'s styles! \`ServerStyleSheet\` is a one off instance for each server-side render cycle.

- Are you trying to reuse it across renders?
- Are you accidentally calling collectStyles twice?

`, 3: `Streaming SSR is only supported in a Node.js environment; Please do not try to call this method in the browser.

`, 4: `The \`StyleSheetManager\` expects a valid target or sheet prop!

- Does this error occur on the client and is your target falsy?
- Does this error occur on the server and is the sheet falsy?

`, 5: `The clone method cannot be used on the client!

- Are you running in a client-like environment on the server?
- Are you trying to run SSR on the client?

`, 6: `Trying to insert a new style tag, but the given Node is unmounted!

- Are you using a custom target that isn't mounted?
- Does your document not have a valid head element?
- Have you accidentally removed a style tag manually?

`, 7: 'ThemeProvider: Please return an object from your "theme" prop function, e.g.\n\n```js\ntheme={() => ({})}\n```\n\n', 8: `ThemeProvider: Please make your "theme" prop an object.

`, 9: "Missing document `<head>`\n\n", 10: `Cannot find a StyleSheet instance. Usually this happens if there are multiple copies of styled-components loaded at once. Check out this issue for how to troubleshoot and fix the common cases where this situation can happen: https://github.com/styled-components/styled-components/issues/1941#issuecomment-417862021

`, 11: `_This error was replaced with a dev-time warning, it will be deleted for v4 final._ [createGlobalStyle] received children which will not be rendered. Please use the component without passing children elements.

`, 12: "It seems you are interpolating a keyframe declaration (%s) into an untagged string. This was supported in styled-components v3, but is not longer supported in v4 as keyframes are now injected on-demand. Please wrap your string in the css\\`\\` helper which ensures the styles are injected correctly. See https://www.styled-components.com/docs/api#css\n\n", 13: `%s is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.

`, 14: `ThemeProvider: "theme" prop is required.

`, 15: "A stylis plugin has been supplied that is not named. We need a name for each plugin to be able to prevent styling collisions between different stylis configurations within the same app. Before you pass your plugin to `<StyleSheetManager stylisPlugins={[]}>`, please make sure each plugin is uniquely-named, e.g.\n\n```js\nObject.defineProperty(importedPlugin, 'name', { value: 'some-unique-name' });\n```\n\n", 16: `Reached the limit of how many styled components may be created at group %s.
You may only create up to 1,073,741,824 components. If you're creating components dynamically,
as for instance in your render method then you may be running into this limitation.

`, 17: `CSSStyleSheet could not be found on HTMLStyleElement.
Has styled-components' style tag been unmounted or altered by another script?
`, 18: "ThemeProvider: Please make sure your useTheme hook is within a `<ThemeProvider>`" } : {};
function us() {
  for (var e = [], t = 0; t < arguments.length; t++)
    e[t] = arguments[t];
  for (var r = e[0], n = [], o = 1, a = e.length; o < a; o += 1)
    n.push(e[o]);
  return n.forEach(function(i) {
    r = r.replace(/%[a-z]/, i);
  }), r;
}
function Ne(e) {
  for (var t = [], r = 1; r < arguments.length; r++)
    t[r - 1] = arguments[r];
  return process.env.NODE_ENV === "production" ? new Error("An error occurred. See https://github.com/styled-components/styled-components/blob/main/packages/styled-components/src/utils/errors.md#".concat(e, " for more information.").concat(t.length > 0 ? " Args: ".concat(t.join(", ")) : "")) : new Error(us.apply(void 0, xe([ls[e]], t, !1)).trim());
}
var fs = function() {
  function e(t) {
    this.groupSizes = new Uint32Array(512), this.length = 512, this.tag = t;
  }
  return e.prototype.indexOfGroup = function(t) {
    for (var r = 0, n = 0; n < t; n++)
      r += this.groupSizes[n];
    return r;
  }, e.prototype.insertRules = function(t, r) {
    if (t >= this.groupSizes.length) {
      for (var n = this.groupSizes, o = n.length, a = o; t >= a; )
        if ((a <<= 1) < 0)
          throw Ne(16, "".concat(t));
      this.groupSizes = new Uint32Array(a), this.groupSizes.set(n), this.length = a;
      for (var i = o; i < a; i++)
        this.groupSizes[i] = 0;
    }
    for (var s = this.indexOfGroup(t + 1), l = (i = 0, r.length); i < l; i++)
      this.tag.insertRule(s, r[i]) && (this.groupSizes[t]++, s++);
  }, e.prototype.clearGroup = function(t) {
    if (t < this.length) {
      var r = this.groupSizes[t], n = this.indexOfGroup(t), o = n + r;
      this.groupSizes[t] = 0;
      for (var a = n; a < o; a++)
        this.tag.deleteRule(n);
    }
  }, e.prototype.getGroup = function(t) {
    var r = "";
    if (t >= this.length || this.groupSizes[t] === 0)
      return r;
    for (var n = this.groupSizes[t], o = this.indexOfGroup(t), a = o + n, i = o; i < a; i++)
      r += "".concat(this.tag.getRule(i)).concat(nr);
    return r;
  }, e;
}(), ps = 1 << 30, at = /* @__PURE__ */ new Map(), lt = /* @__PURE__ */ new Map(), it = 1, Qe = function(e) {
  if (at.has(e))
    return at.get(e);
  for (; lt.has(it); )
    it++;
  var t = it++;
  if (process.env.NODE_ENV !== "production" && ((0 | t) < 0 || t > ps))
    throw Ne(16, "".concat(t));
  return at.set(e, t), lt.set(t, e), t;
}, ds = function(e, t) {
  it = t + 1, at.set(e, t), lt.set(t, e);
}, ys = "style[".concat(Pe, "][").concat(Ln, '="').concat(bt, '"]'), hs = new RegExp("^".concat(Pe, '\\.g(\\d+)\\[id="([\\w\\d-]+)"\\].*?"([^"]*)')), ms = function(e, t, r) {
  for (var n, o = r.split(","), a = 0, i = o.length; a < i; a++)
    (n = o[a]) && e.registerName(t, n);
}, gs = function(e, t) {
  for (var r, n = ((r = t.textContent) !== null && r !== void 0 ? r : "").split(nr), o = [], a = 0, i = n.length; a < i; a++) {
    var s = n[a].trim();
    if (s) {
      var l = s.match(hs);
      if (l) {
        var h = 0 | parseInt(l[1], 10), p = l[2];
        h !== 0 && (ds(p, h), ms(e, p, l[3]), e.getTag().insertRules(h, o)), o.length = 0;
      } else
        o.push(s);
    }
  }
}, Nr = function(e) {
  for (var t = document.querySelectorAll(ys), r = 0, n = t.length; r < n; r++) {
    var o = t[r];
    o && o.getAttribute(Pe) !== Nn && (gs(e, o), o.parentNode && o.parentNode.removeChild(o));
  }
};
function vs() {
  return typeof __webpack_nonce__ < "u" ? __webpack_nonce__ : null;
}
var Bn = function(e) {
  var t = document.head, r = e || t, n = document.createElement("style"), o = function(s) {
    var l = Array.from(s.querySelectorAll("style[".concat(Pe, "]")));
    return l[l.length - 1];
  }(r), a = o !== void 0 ? o.nextSibling : null;
  n.setAttribute(Pe, Nn), n.setAttribute(Ln, bt);
  var i = vs();
  return i && n.setAttribute("nonce", i), r.insertBefore(n, a), n;
}, bs = function() {
  function e(t) {
    this.element = Bn(t), this.element.appendChild(document.createTextNode("")), this.sheet = function(r) {
      if (r.sheet)
        return r.sheet;
      for (var n = document.styleSheets, o = 0, a = n.length; o < a; o++) {
        var i = n[o];
        if (i.ownerNode === r)
          return i;
      }
      throw Ne(17);
    }(this.element), this.length = 0;
  }
  return e.prototype.insertRule = function(t, r) {
    try {
      return this.sheet.insertRule(r, t), this.length++, !0;
    } catch {
      return !1;
    }
  }, e.prototype.deleteRule = function(t) {
    this.sheet.deleteRule(t), this.length--;
  }, e.prototype.getRule = function(t) {
    var r = this.sheet.cssRules[t];
    return r && r.cssText ? r.cssText : "";
  }, e;
}(), _s = function() {
  function e(t) {
    this.element = Bn(t), this.nodes = this.element.childNodes, this.length = 0;
  }
  return e.prototype.insertRule = function(t, r) {
    if (t <= this.length && t >= 0) {
      var n = document.createTextNode(r);
      return this.element.insertBefore(n, this.nodes[t] || null), this.length++, !0;
    }
    return !1;
  }, e.prototype.deleteRule = function(t) {
    this.element.removeChild(this.nodes[t]), this.length--;
  }, e.prototype.getRule = function(t) {
    return t < this.length ? this.nodes[t].textContent : "";
  }, e;
}(), Ps = function() {
  function e(t) {
    this.rules = [], this.length = 0;
  }
  return e.prototype.insertRule = function(t, r) {
    return t <= this.length && (this.rules.splice(t, 0, r), this.length++, !0);
  }, e.prototype.deleteRule = function(t) {
    this.rules.splice(t, 1), this.length--;
  }, e.prototype.getRule = function(t) {
    return t < this.length ? this.rules[t] : "";
  }, e;
}(), Lr = ct, ws = { isServer: !ct, useCSSOMInjection: !Xi }, qn = function() {
  function e(t, r, n) {
    t === void 0 && (t = Me), r === void 0 && (r = {});
    var o = this;
    this.options = ee(ee({}, ws), t), this.gs = r, this.names = new Map(n), this.server = !!t.isServer, !this.server && ct && Lr && (Lr = !1, Nr(this)), ar(this, function() {
      return function(a) {
        for (var i = a.getTag(), s = i.length, l = "", h = function(d) {
          var m = function($) {
            return lt.get($);
          }(d);
          if (m === void 0)
            return "continue";
          var g = a.names.get(m), R = i.getGroup(d);
          if (g === void 0 || !g.size || R.length === 0)
            return "continue";
          var b = "".concat(Pe, ".g").concat(d, '[id="').concat(m, '"]'), j = "";
          g !== void 0 && g.forEach(function($) {
            $.length > 0 && (j += "".concat($, ","));
          }), l += "".concat(R).concat(b, '{content:"').concat(j, '"}').concat(nr);
        }, p = 0; p < s; p++)
          h(p);
        return l;
      }(o);
    });
  }
  return e.registerId = function(t) {
    return Qe(t);
  }, e.prototype.rehydrate = function() {
    !this.server && ct && Nr(this);
  }, e.prototype.reconstructWithOptions = function(t, r) {
    return r === void 0 && (r = !0), new e(ee(ee({}, this.options), t), this.gs, r && this.names || void 0);
  }, e.prototype.allocateGSInstance = function(t) {
    return this.gs[t] = (this.gs[t] || 0) + 1;
  }, e.prototype.getTag = function() {
    return this.tag || (this.tag = (t = function(r) {
      var n = r.useCSSOMInjection, o = r.target;
      return r.isServer ? new Ps(o) : n ? new bs(o) : new _s(o);
    }(this.options), new fs(t)));
    var t;
  }, e.prototype.hasNameForId = function(t, r) {
    return this.names.has(t) && this.names.get(t).has(r);
  }, e.prototype.registerName = function(t, r) {
    if (Qe(t), this.names.has(t))
      this.names.get(t).add(r);
    else {
      var n = /* @__PURE__ */ new Set();
      n.add(r), this.names.set(t, n);
    }
  }, e.prototype.insertRules = function(t, r, n) {
    this.registerName(t, r), this.getTag().insertRules(Qe(t), n);
  }, e.prototype.clearNames = function(t) {
    this.names.has(t) && this.names.get(t).clear();
  }, e.prototype.clearRules = function(t) {
    this.getTag().clearGroup(Qe(t)), this.clearNames(t);
  }, e.prototype.clearTag = function() {
    this.tag = void 0;
  }, e;
}(), Ss = /&/g, Os = /^\s*\/\/.*$/gm;
function Xn(e, t) {
  return e.map(function(r) {
    return r.type === "rule" && (r.value = "".concat(t, " ").concat(r.value), r.value = r.value.replaceAll(",", ",".concat(t, " ")), r.props = r.props.map(function(n) {
      return "".concat(t, " ").concat(n);
    })), Array.isArray(r.children) && r.type !== "@keyframes" && (r.children = Xn(r.children, t)), r;
  });
}
function Es(e) {
  var t, r, n, o = e === void 0 ? Me : e, a = o.options, i = a === void 0 ? Me : a, s = o.plugins, l = s === void 0 ? _t : s, h = function(m, g, R) {
    return R.startsWith(r) && R.endsWith(r) && R.replaceAll(r, "").length > 0 ? ".".concat(t) : m;
  }, p = l.slice();
  p.push(function(m) {
    m.type === ht && m.value.includes("&") && (m.props[0] = m.props[0].replace(Ss, r).replace(n, h));
  }), i.prefix && p.push(Bi), p.push(Vi);
  var d = function(m, g, R, b) {
    g === void 0 && (g = ""), R === void 0 && (R = ""), b === void 0 && (b = "&"), t = b, r = g, n = new RegExp("\\".concat(r, "\\b"), "g");
    var j = m.replace(Os, ""), $ = Hi(R || g ? "".concat(R, " ").concat(g, " { ").concat(j, " }") : j);
    i.namespace && ($ = Xn($, i.namespace));
    var D = [];
    return st($, Yi(p.concat(Wi(function(_) {
      return D.push(_);
    })))), D;
  };
  return d.hash = l.length ? l.reduce(function(m, g) {
    return g.name || Ne(15), ge(m, g.name);
  }, Un).toString() : "", d;
}
var Ts = new qn(), Bt = Es(), Gn = ye.createContext({ shouldForwardProp: void 0, styleSheet: Ts, stylis: Bt });
Gn.Consumer;
ye.createContext(void 0);
function Ur() {
  return ro(Gn);
}
var zr = function() {
  function e(t, r) {
    var n = this;
    this.inject = function(o, a) {
      a === void 0 && (a = Bt);
      var i = n.name + a.hash;
      o.hasNameForId(n.id, i) || o.insertRules(n.id, i, a(n.rules, i, "@keyframes"));
    }, this.name = t, this.id = "sc-keyframes-".concat(t), this.rules = r, ar(this, function() {
      throw Ne(12, String(n.name));
    });
  }
  return e.prototype.getName = function(t) {
    return t === void 0 && (t = Bt), this.name + t.hash;
  }, e;
}(), Cs = function(e) {
  return e >= "A" && e <= "Z";
};
function Hr(e) {
  for (var t = "", r = 0; r < e.length; r++) {
    var n = e[r];
    if (r === 1 && n === "-" && e[0] === "-")
      return e;
    Cs(n) ? t += "-" + n.toLowerCase() : t += n;
  }
  return t.startsWith("ms-") ? "-" + t : t;
}
var Kn = function(e) {
  return e == null || e === !1 || e === "";
}, Jn = function(e) {
  var t, r, n = [];
  for (var o in e) {
    var a = e[o];
    e.hasOwnProperty(o) && !Kn(a) && (Array.isArray(a) && a.isCss || $e(a) ? n.push("".concat(Hr(o), ":"), a, ";") : je(a) ? n.push.apply(n, xe(xe(["".concat(o, " {")], Jn(a), !1), ["}"], !1)) : n.push("".concat(Hr(o), ": ").concat((t = o, (r = a) == null || typeof r == "boolean" || r === "" ? "" : typeof r != "number" || r === 0 || t in qi || t.startsWith("--") ? String(r).trim() : "".concat(r, "px")), ";")));
  }
  return n;
};
function _e(e, t, r, n) {
  if (Kn(e))
    return [];
  if (or(e))
    return [".".concat(e.styledComponentId)];
  if ($e(e)) {
    if (!$e(a = e) || a.prototype && a.prototype.isReactComponent || !t)
      return [e];
    var o = e(t);
    return process.env.NODE_ENV === "production" || typeof o != "object" || Array.isArray(o) || o instanceof zr || je(o) || o === null || console.error("".concat(Hn(e), " is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.")), _e(o, t, r, n);
  }
  var a;
  return e instanceof zr ? r ? (e.inject(r, n), [e.getName(n)]) : [e] : je(e) ? Jn(e) : Array.isArray(e) ? Array.prototype.concat.apply(_t, e.map(function(i) {
    return _e(i, t, r, n);
  })) : [e.toString()];
}
function As(e) {
  for (var t = 0; t < e.length; t += 1) {
    var r = e[t];
    if ($e(r) && !or(r))
      return !1;
  }
  return !0;
}
var Rs = zn(bt), xs = function() {
  function e(t, r, n) {
    this.rules = t, this.staticRulesId = "", this.isStatic = process.env.NODE_ENV === "production" && (n === void 0 || n.isStatic) && As(t), this.componentId = r, this.baseHash = ge(Rs, r), this.baseStyle = n, qn.registerId(r);
  }
  return e.prototype.generateAndInjectStyles = function(t, r, n) {
    var o = this.baseStyle ? this.baseStyle.generateAndInjectStyles(t, r, n) : "";
    if (this.isStatic && !n.hash)
      if (this.staticRulesId && r.hasNameForId(this.componentId, this.staticRulesId))
        o = ve(o, this.staticRulesId);
      else {
        var a = Dr(_e(this.rules, t, r, n)), i = Yt(ge(this.baseHash, a) >>> 0);
        if (!r.hasNameForId(this.componentId, i)) {
          var s = n(a, ".".concat(i), void 0, this.componentId);
          r.insertRules(this.componentId, i, s);
        }
        o = ve(o, i), this.staticRulesId = i;
      }
    else {
      for (var l = ge(this.baseHash, n.hash), h = "", p = 0; p < this.rules.length; p++) {
        var d = this.rules[p];
        if (typeof d == "string")
          h += d, process.env.NODE_ENV !== "production" && (l = ge(l, d));
        else if (d) {
          var m = Dr(_e(d, t, r, n));
          l = ge(l, m + p), h += m;
        }
      }
      if (h) {
        var g = Yt(l >>> 0);
        r.hasNameForId(this.componentId, g) || r.insertRules(this.componentId, g, n(h, ".".concat(g), void 0, this.componentId)), o = ve(o, g);
      }
    }
    return o;
  }, e;
}(), Zn = ye.createContext(void 0);
Zn.Consumer;
var $t = {}, Fr = /* @__PURE__ */ new Set();
function ks(e, t, r) {
  var n = or(e), o = e, a = !Mt(e), i = t.attrs, s = i === void 0 ? _t : i, l = t.componentId, h = l === void 0 ? function(P, u) {
    var O = typeof P != "string" ? "sc" : kr(P);
    $t[O] = ($t[O] || 0) + 1;
    var w = "".concat(O, "-").concat(es(bt + O + $t[O]));
    return u ? "".concat(u, "-").concat(w) : w;
  }(t.displayName, t.parentComponentId) : l, p = t.displayName, d = p === void 0 ? function(P) {
    return Mt(P) ? "styled.".concat(P) : "Styled(".concat(Hn(P), ")");
  }(e) : p, m = t.displayName && t.componentId ? "".concat(kr(t.displayName), "-").concat(t.componentId) : t.componentId || h, g = n && o.attrs ? o.attrs.concat(s).filter(Boolean) : s, R = t.shouldForwardProp;
  if (n && o.shouldForwardProp) {
    var b = o.shouldForwardProp;
    if (t.shouldForwardProp) {
      var j = t.shouldForwardProp;
      R = function(P, u) {
        return b(P, u) && j(P, u);
      };
    } else
      R = b;
  }
  var $ = new xs(r, m, n ? o.componentStyle : void 0);
  function D(P, u) {
    return function(O, w, ne) {
      var ie = O.attrs, q = O.componentStyle, we = O.defaultProps, Le = O.foldedComponentIds, Se = O.styledComponentId, pe = O.target, Ue = ye.useContext(Zn), oe = Ur(), se = O.shouldForwardProp || oe.shouldForwardProp;
      process.env.NODE_ENV !== "production" && cr(Se);
      var he = Ki(w, Ue, we) || Me, te = function(k, E, T) {
        for (var x, I = ee(ee({}, E), { className: void 0, theme: T }), M = 0; M < k.length; M += 1) {
          var Y = $e(x = k[M]) ? x(I) : x;
          for (var f in Y)
            I[f] = f === "className" ? ve(I[f], Y[f]) : f === "style" ? ee(ee({}, I[f]), Y[f]) : Y[f];
        }
        return E.className && (I.className = ve(I.className, E.className)), I;
      }(ie, w, he), c = te.as || pe, y = {};
      for (var v in te)
        te[v] === void 0 || v[0] === "$" || v === "as" || v === "theme" && te.theme === he || (v === "forwardedAs" ? y.as = te.forwardedAs : se && !se(v, c) || (y[v] = te[v], se || process.env.NODE_ENV !== "development" || Ai(v) || Fr.has(v) || !Vt.has(c) || (Fr.add(v), console.warn('styled-components: it looks like an unknown prop "'.concat(v, '" is being sent through to the DOM, which will likely trigger a React console error. If you would like automatic filtering of unknown props, you can opt-into that behavior via `<StyleSheetManager shouldForwardProp={...}>` (connect an API like `@emotion/is-prop-valid`) or consider using transient props (`$` prefix for automatic filtering.)')))));
      var C = function(k, E) {
        var T = Ur(), x = k.generateAndInjectStyles(E, T.styleSheet, T.stylis);
        return process.env.NODE_ENV !== "production" && cr(x), x;
      }(q, te);
      process.env.NODE_ENV !== "production" && O.warnTooManyClasses && O.warnTooManyClasses(C);
      var S = ve(Le, Se);
      return C && (S += " " + C), te.className && (S += " " + te.className), y[Mt(c) && !Vt.has(c) ? "class" : "className"] = S, y.ref = ne, to(c, y);
    }(_, P, u);
  }
  D.displayName = d;
  var _ = ye.forwardRef(D);
  return _.attrs = g, _.componentStyle = $, _.displayName = d, _.shouldForwardProp = R, _.foldedComponentIds = n ? ve(o.foldedComponentIds, o.styledComponentId) : "", _.styledComponentId = m, _.target = n ? o.target : e, Object.defineProperty(_, "defaultProps", { get: function() {
    return this._foldedDefaultProps;
  }, set: function(P) {
    this._foldedDefaultProps = n ? function(u) {
      for (var O = [], w = 1; w < arguments.length; w++)
        O[w - 1] = arguments[w];
      for (var ne = 0, ie = O; ne < ie.length; ne++)
        Wt(u, ie[ne], !0);
      return u;
    }({}, o.defaultProps, P) : P;
  } }), process.env.NODE_ENV !== "production" && (Gi(d, m), _.warnTooManyClasses = /* @__PURE__ */ function(P, u) {
    var O = {}, w = !1;
    return function(ne) {
      if (!w && (O[ne] = !0, Object.keys(O).length >= 200)) {
        var ie = u ? ' with the id of "'.concat(u, '"') : "";
        console.warn("Over ".concat(200, " classes were generated for component ").concat(P).concat(ie, `.
`) + `Consider using the attrs method, together with a style object for frequently changed styles.
Example:
  const Component = styled.div.attrs(props => ({
    style: {
      background: props.background,
    },
  }))\`width: 100%;\`

  <Component />`), w = !0, O = {};
      }
    };
  }(d, m)), ar(_, function() {
    return ".".concat(_.styledComponentId);
  }), a && Wn(_, e, { attrs: !0, componentStyle: !0, displayName: !0, foldedComponentIds: !0, shouldForwardProp: !0, styledComponentId: !0, target: !0 }), _;
}
function Vr(e, t) {
  for (var r = [e[0]], n = 0, o = t.length; n < o; n += 1)
    r.push(t[n], e[n + 1]);
  return r;
}
var Yr = function(e) {
  return Object.assign(e, { isCss: !0 });
};
function Is(e) {
  for (var t = [], r = 1; r < arguments.length; r++)
    t[r - 1] = arguments[r];
  if ($e(e) || je(e))
    return Yr(_e(Vr(_t, xe([e], t, !0))));
  var n = e;
  return t.length === 0 && n.length === 1 && typeof n[0] == "string" ? _e(n) : Yr(_e(Vr(n, t)));
}
function qt(e, t, r) {
  if (r === void 0 && (r = Me), !t)
    throw Ne(1, t);
  var n = function(o) {
    for (var a = [], i = 1; i < arguments.length; i++)
      a[i - 1] = arguments[i];
    return e(t, r, Is.apply(void 0, xe([o], a, !1)));
  };
  return n.attrs = function(o) {
    return qt(e, t, ee(ee({}, r), { attrs: Array.prototype.concat(r.attrs, o).filter(Boolean) }));
  }, n.withConfig = function(o) {
    return qt(e, t, ee(ee({}, r), o));
  }, n;
}
var Qn = function(e) {
  return qt(ks, e);
}, ir = Qn;
Vt.forEach(function(e) {
  ir[e] = Qn(e);
});
process.env.NODE_ENV !== "production" && typeof navigator < "u" && navigator.product === "ReactNative" && console.warn(`It looks like you've imported 'styled-components' on React Native.
Perhaps you're looking to import 'styled-components/native'?
Read more about this at https://www.styled-components.com/docs/basics#react-native`);
var et = "__sc-".concat(Pe, "__");
process.env.NODE_ENV !== "production" && process.env.NODE_ENV !== "test" && typeof window < "u" && (window[et] || (window[et] = 0), window[et] === 1 && console.warn(`It looks like there are several instances of 'styled-components' initialized in this application. This may cause dynamic styles to not render properly, errors during the rehydration process, a missing theme prop, and makes your application bigger without good reason.

See https://s-c.sh/2BAXzed for more info.`), window[et] += 1);
function Ls(e) {
  const {
    isPlay: t,
    setIsPlay: r,
    url: n,
    width: o,
    height: a
  } = e, i = Wr(null), [s, l] = Oe(!1), [h, p] = Oe(!1), [d, m] = Oe(0), [g, R] = Oe(0), [b, j] = Oe(1), [$, D] = Oe(!1), _ = () => {
    r(!1);
  }, P = () => {
    p(!h);
  }, u = () => {
    D(!0);
  }, O = (q) => {
    m(parseFloat(q.target.value));
  }, w = (q) => {
    D(!1), i.current && i.current.seekTo(parseFloat(q.target.value));
  };
  return /* @__PURE__ */ me(Ms, { children: /* @__PURE__ */ eo($s, { children: [
    /* @__PURE__ */ me(
      Tr,
      {
        width: o,
        height: a,
        ref: i,
        url: n,
        className: "hogak-player",
        playing: t,
        controls: !1,
        onEnded: _,
        onReady: () => l(!0),
        onDuration: (q) => {
          console.log("onDuration", q), R(q);
        },
        onProgress: (q) => {
          console.log("onProgress", q), $ || m(q.played);
        },
        volume: b,
        pip: h
      }
    ),
    /* @__PURE__ */ me(lr, { seconds: g * d }),
    " / ",
    /* @__PURE__ */ me(lr, { seconds: g }),
    /* @__PURE__ */ me(
      "input",
      {
        style: { width: "100%" },
        type: "range",
        min: 0,
        max: 0.999999,
        step: "any",
        value: d,
        onMouseDown: u,
        onChange: O,
        onMouseUp: w
      }
    ),
    /* @__PURE__ */ me("input", { type: "range", min: 0, max: 1, step: "any", value: b, onChange: (q) => j(parseFloat(q.target.value)) }),
    Tr.canEnablePIP(n) && /* @__PURE__ */ me("button", { onClick: P, children: h ? "PIP 비활성화" : "PIP 활성화" })
  ] }) });
}
const Ms = ir.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`, $s = ir.div`
  padding: 20px;
  z-index: 2;
`;
export {
  Ls as H,
  no as g,
  ea as p,
  ft as u
};
